// @forge/util/packages/ari/people-data/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);

// src/people-data/index.ts
var people_data_exports = {};
__export(people_data_exports, {
  PeopleDataCostCenterAri: () => PeopleDataCostCenterAri,
  PeopleDataJobProfileAri: () => PeopleDataJobProfileAri,
  PeopleDataPositionAri: () => PeopleDataPositionAri,
  PeopleDataSupervisoryOrgAri: () => PeopleDataSupervisoryOrgAri,
  PeopleDataWorkerAri: () => PeopleDataWorkerAri
});
module.exports = __toCommonJS(people_data_exports);

// src/errors.ts
var ValidationError = class extends Error {
};

// src/core/rules/cloud-id.ts
function validateCloudId(cloudId, format = new RegExp("^[a-zA-Z0-9\\-]*$")) {
  if (!cloudId.match(format))
    throw new ValidationError(`Invalid cloud ID, expected ID of format ${format}.`);
}

// src/core/rules/platform-qualifier.ts
function validatePlatformQualifier(qualifier) {
  if (qualifier !== "cloud" && qualifier !== "third-party")
    throw new ValidationError(`Identifier must have a qualifier of 'cloud' or 'third-party'. Received: ${qualifier}`);
}

// src/core/rules/qualifier.ts
function validateAtiQualifier(qualifier) {
  if (qualifier !== "ati")
    throw new ValidationError(`ATI must have a qualifier of 'ati'. Received: ${qualifier}`);
}
function validateAriQualifier(qualifier) {
  if (qualifier !== "ari")
    throw new ValidationError(`ARI must have a qualifier of 'ari'. Received: ${qualifier}`);
}

// src/core/rules/resource-id.ts
function validateResourceId(id, format = new RegExp("[!a-zA-Z0-9\\-_.~@:{}=]+(/[!a-zA-Z0-9\\-_.~@:{}=]+)*" /* ANY_RESOURCE_ID */), key = "resourceId") {
  let formatWithCarets = new RegExp(`^${format.source}$`);
  if (!id.match(formatWithCarets))
    throw new ValidationError(`Invalid ${key} - ${id}, expected ID of format ${formatWithCarets}.`);
}

// src/core/rules/resource-id-segments.ts
function validateResourceIdSegments(resourceIdSegmentValues, resourceIdSegmentFormats) {
  Object.entries(resourceIdSegmentValues).forEach(([resourceIdKey, resourceIdValue]) => {
    validateResourceId(resourceIdValue, resourceIdSegmentFormats == null ? void 0 : resourceIdSegmentFormats[resourceIdKey], resourceIdKey);
  });
}

// src/core/rules/resource-owner.ts
function validateResourceOwner(owner, expectedResourceOwner) {
  if (expectedResourceOwner && owner !== expectedResourceOwner)
    throw new ValidationError(`Invalid resource owner - ${owner}, expected ${expectedResourceOwner}.`);
}

// src/core/rules/resource-type.ts
function validateResourceType(type, expectedResourceType) {
  if (expectedResourceType && type !== expectedResourceType)
    throw new ValidationError(`Invalid resource type - ${type}, expected ${expectedResourceType}.`);
}

// src/core/parser/base.ts
var NUMBER_OF_BASE_SEGMENTS = 5, SEGMENT_SEPARATOR = ":", BaseParser = class {
  static getIdentifierSegments(inputStr, numberOfSegmentsRequired) {
    let allSegments = inputStr.split(SEGMENT_SEPARATOR);
    if (allSegments.length < numberOfSegmentsRequired)
      throw new ValidationError(`Input string must have ${numberOfSegmentsRequired} segments.`);
    if (allSegments.length > numberOfSegmentsRequired) {
      let segments = allSegments.slice(0, NUMBER_OF_BASE_SEGMENTS - 1), segmentsForResourceId = allSegments.slice(NUMBER_OF_BASE_SEGMENTS - 1, allSegments.length);
      return [...segments, segmentsForResourceId.join(SEGMENT_SEPARATOR)];
    }
    return allSegments;
  }
};

// src/core/parser/ari.ts
var NUMBER_OF_ARI_SEGMENTS = 5, AriParser = class extends BaseParser {
  static fromString(maybeAri, ariProperties) {
    let segments = this.getIdentifierSegments(maybeAri, NUMBER_OF_ARI_SEGMENTS), [qualifier, platformQualifier, resourceOwner, cloudId, resourceTypeAndId] = segments, [resourceType, ...resourceIdSegments] = resourceTypeAndId.split("/"), maybeAriDerivedProperties = {
      qualifier,
      platformQualifier,
      cloudId,
      resourceOwner,
      resourceType,
      resourceId: resourceIdSegments.join("/"),
      resourceIdSegmentValues: ariProperties != null && ariProperties.resourceIdSlug ? this.ariSegmentValuesFromSlug(
        ariProperties.resourceIdSlug,
        resourceIdSegments.join("/"),
        Object.keys(ariProperties.resourceIdSegmentFormats)
      ) : {}
    };
    return this.fromOpts(maybeAriDerivedProperties, ariProperties);
  }
  static fromOpts(maybeAriOpts, ariProperties) {
    let {
      qualifier = "ari",
      platformQualifier = "cloud",
      resourceOwner,
      cloudId,
      resourceId,
      resourceIdSegmentValues,
      resourceType
    } = maybeAriOpts;
    return validateAriQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateCloudId(maybeAriOpts.cloudId || "", ariProperties == null ? void 0 : ariProperties.cloudId), validateResourceOwner(resourceOwner, ariProperties == null ? void 0 : ariProperties.resourceOwner), validateResourceType(resourceType, ariProperties == null ? void 0 : ariProperties.resourceType), validateResourceIdSegments(resourceIdSegmentValues, ariProperties == null ? void 0 : ariProperties.resourceIdSegmentFormats), {
      qualifier,
      resourceOwner,
      resourceType,
      platformQualifier,
      cloudId,
      resourceId,
      resourceIdSegmentValues
    };
  }
  static ariSegmentValuesFromSlug(slug, incomingSegments, expectedKeys) {
    let regexpResult = new RegExp("^" + slug.replace(/\{(.*?)\}/g, "(?<$1>.*?)") + "$").exec(incomingSegments);
    if (!regexpResult)
      throw new ValidationError(`Segment '${incomingSegments}' don't match expected slug: ${slug}`);
    return expectedKeys.forEach((expectedKey) => {
      var _a;
      if (!((_a = regexpResult.groups) != null && _a[expectedKey]))
        throw new ValidationError(`No value supplied for '${expectedKey}' based on slug ${slug}`);
    }), regexpResult.groups || {};
  }
};

// src/core/parser/ati.ts
var NUMBER_OF_ATI_SEGMENTS = 4, AtiParser = class extends BaseParser {
  static fromString(inputStr, atiOpts) {
    let segments = this.getIdentifierSegments(inputStr, NUMBER_OF_ATI_SEGMENTS), [qualifier = "ati", platformQualifier = "cloud", resourceOwner, resourceType] = segments;
    return validateAtiQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateResourceOwner(resourceOwner, atiOpts == null ? void 0 : atiOpts.resourceOwner), validateResourceType(resourceType, atiOpts == null ? void 0 : atiOpts.resourceType), { platformQualifier, resourceOwner, resourceType };
  }
  static fromOpts(maybeAtiOpts, atiOpts) {
    let { qualifier, platformQualifier, resourceOwner, resourceType } = maybeAtiOpts;
    return validateAtiQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateResourceOwner(resourceOwner, atiOpts == null ? void 0 : atiOpts.resourceOwner), validateResourceType(resourceType, atiOpts == null ? void 0 : atiOpts.resourceType), {
      platformQualifier,
      resourceOwner,
      resourceType
    };
  }
};

// src/ati.ts
var Ati = class _Ati {
  constructor(platformQualifier, resourceOwner, resourceType) {
    this._platformQualifier = platformQualifier, this._resourceOwner = resourceOwner, this._resourceType = resourceType;
  }
  get platformQualifier() {
    return this._platformQualifier;
  }
  get resourceOwner() {
    return this._resourceOwner;
  }
  get resourceType() {
    return this._resourceType;
  }
  static create(opts) {
    return new _Ati(opts.platformQualifier || "cloud", opts.resourceOwner, opts.resourceType);
  }
  static parse(maybeAti, atiOpts) {
    let opts = AtiParser.fromString(maybeAti, atiOpts);
    return new _Ati(opts.platformQualifier, opts.resourceOwner, opts.resourceType);
  }
  toString() {
    return `ati:${this.platformQualifier}:${this.resourceOwner}:${this.resourceType}`;
  }
  toJSON() {
    return this.toString();
  }
  toOpts() {
    return {
      platformQualifier: this._platformQualifier,
      resourceOwner: this._resourceOwner,
      resourceType: this._resourceType
    };
  }
};

// src/core/ari.ts
var Ari = class {
  constructor(opts) {
    this._ati = Ati.create({
      platformQualifier: opts.platformQualifier || "cloud",
      resourceOwner: opts.resourceOwner,
      resourceType: opts.resourceType
    }), this._cloudId = opts.cloudId !== "" ? opts.cloudId : void 0, this._resourceId = opts.resourceId;
  }
  get platformQualifier() {
    return this._ati.platformQualifier;
  }
  get cloudId() {
    return this._cloudId;
  }
  get resourceOwner() {
    return this._ati.resourceOwner;
  }
  get resourceType() {
    return this._ati.resourceType;
  }
  get resourceId() {
    return this._resourceId;
  }
  get ati() {
    return this._ati;
  }
  equals(other) {
    return this.toString() === other.toString();
  }
  toString() {
    return `ari:${this.platformQualifier}:${this.resourceOwner}:${this.cloudId || ""}:${this.resourceType}/${this.resourceId}`;
  }
  toJSON() {
    return this.toString();
  }
  toOpts() {
    return {
      platformQualifier: this.platformQualifier,
      resourceOwner: this.resourceOwner,
      cloudId: this.cloudId,
      resourceType: this.resourceType,
      resourceId: this.resourceId
    };
  }
};

// src/any-ari.ts
var AnyAri = class _AnyAri extends Ari {
  constructor(opts) {
    super(opts);
  }
  static create(ariOpts, ariStaticOpts) {
    let ariOptsWithDefaults = { ...ariOpts, resourceIdSegmentValues: {} }, validatedOpts = AriParser.fromOpts(ariOptsWithDefaults, ariStaticOpts);
    return new _AnyAri(validatedOpts);
  }
  static parse(maybeAri, ariStaticOpts) {
    let validatedOpts = AriParser.fromString(maybeAri, ariStaticOpts);
    return new _AnyAri(validatedOpts);
  }
  static check(maybeAri) {
    try {
      return _AnyAri.parse(maybeAri.toString()), !0;
    } catch (err) {
      return !1;
    }
  }
  asAnyAri() {
    return this;
  }
};

// src/core/registered-ari.ts
var RegisteredAri = class extends Ari {
  static check(maybeAri) {
    try {
      return this.parse(maybeAri.toString()), !0;
    } catch (err) {
      return !1;
    }
  }
  asAnyAri() {
    return AnyAri.parse(this.toString());
  }
};

// src/people-data/cost-center/types.ts
var PeopleDataCostCenterAriResourceOwner = "people-data", PeopleDataCostCenterAriResourceType = "cost-center";

// src/people-data/cost-center/manifest.ts
var peopleDataCostCenterAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataCostCenterAriResourceOwner,
  resourceType: PeopleDataCostCenterAriResourceType,
  resourceIdSlug: "{costCenterId}",
  resourceIdSegmentFormats: {
    costCenterId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/cost-center/index.ts
var PeopleDataCostCenterAri = class _PeopleDataCostCenterAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._costCenterId = opts.resourceIdSegmentValues.costCenterId;
  }
  get costCenterId() {
    return this._costCenterId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataCostCenterAriStaticOpts.qualifier,
      platformQualifier: peopleDataCostCenterAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataCostCenterAriStaticOpts.resourceOwner,
      resourceType: peopleDataCostCenterAriStaticOpts.resourceType,
      resourceId: `${opts.costCenterId}`,
      resourceIdSegmentValues: {
        costCenterId: opts.costCenterId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataCostCenterAriStaticOpts);
    return new _PeopleDataCostCenterAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataCostCenterAriStaticOpts);
    return new _PeopleDataCostCenterAri(opts);
  }
  getVariables() {
    return {
      costCenterId: this.costCenterId
    };
  }
};

// src/people-data/job-profile/types.ts
var PeopleDataJobProfileAriResourceOwner = "people-data", PeopleDataJobProfileAriResourceType = "job-profile";

// src/people-data/job-profile/manifest.ts
var peopleDataJobProfileAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataJobProfileAriResourceOwner,
  resourceType: PeopleDataJobProfileAriResourceType,
  resourceIdSlug: "{jobProfileId}",
  resourceIdSegmentFormats: {
    jobProfileId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/job-profile/index.ts
var PeopleDataJobProfileAri = class _PeopleDataJobProfileAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._jobProfileId = opts.resourceIdSegmentValues.jobProfileId;
  }
  get jobProfileId() {
    return this._jobProfileId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataJobProfileAriStaticOpts.qualifier,
      platformQualifier: peopleDataJobProfileAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataJobProfileAriStaticOpts.resourceOwner,
      resourceType: peopleDataJobProfileAriStaticOpts.resourceType,
      resourceId: `${opts.jobProfileId}`,
      resourceIdSegmentValues: {
        jobProfileId: opts.jobProfileId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataJobProfileAriStaticOpts);
    return new _PeopleDataJobProfileAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataJobProfileAriStaticOpts);
    return new _PeopleDataJobProfileAri(opts);
  }
  getVariables() {
    return {
      jobProfileId: this.jobProfileId
    };
  }
};

// src/people-data/position/types.ts
var PeopleDataPositionAriResourceOwner = "people-data", PeopleDataPositionAriResourceType = "position";

// src/people-data/position/manifest.ts
var peopleDataPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataPositionAriResourceOwner,
  resourceType: PeopleDataPositionAriResourceType,
  resourceIdSlug: "{positionId}",
  resourceIdSegmentFormats: {
    positionId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/position/index.ts
var PeopleDataPositionAri = class _PeopleDataPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._positionId = opts.resourceIdSegmentValues.positionId;
  }
  get positionId() {
    return this._positionId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataPositionAriStaticOpts.qualifier,
      platformQualifier: peopleDataPositionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataPositionAriStaticOpts.resourceOwner,
      resourceType: peopleDataPositionAriStaticOpts.resourceType,
      resourceId: `${opts.positionId}`,
      resourceIdSegmentValues: {
        positionId: opts.positionId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataPositionAriStaticOpts);
    return new _PeopleDataPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataPositionAriStaticOpts);
    return new _PeopleDataPositionAri(opts);
  }
  getVariables() {
    return {
      positionId: this.positionId
    };
  }
};

// src/people-data/supervisory-org/types.ts
var PeopleDataSupervisoryOrgAriResourceOwner = "people-data", PeopleDataSupervisoryOrgAriResourceType = "supervisory-org";

// src/people-data/supervisory-org/manifest.ts
var peopleDataSupervisoryOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataSupervisoryOrgAriResourceOwner,
  resourceType: PeopleDataSupervisoryOrgAriResourceType,
  resourceIdSlug: "{supervisoryOrgId}",
  resourceIdSegmentFormats: {
    supervisoryOrgId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/supervisory-org/index.ts
var PeopleDataSupervisoryOrgAri = class _PeopleDataSupervisoryOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._supervisoryOrgId = opts.resourceIdSegmentValues.supervisoryOrgId;
  }
  get supervisoryOrgId() {
    return this._supervisoryOrgId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataSupervisoryOrgAriStaticOpts.qualifier,
      platformQualifier: peopleDataSupervisoryOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataSupervisoryOrgAriStaticOpts.resourceOwner,
      resourceType: peopleDataSupervisoryOrgAriStaticOpts.resourceType,
      resourceId: `${opts.supervisoryOrgId}`,
      resourceIdSegmentValues: {
        supervisoryOrgId: opts.supervisoryOrgId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataSupervisoryOrgAriStaticOpts);
    return new _PeopleDataSupervisoryOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataSupervisoryOrgAriStaticOpts);
    return new _PeopleDataSupervisoryOrgAri(opts);
  }
  getVariables() {
    return {
      supervisoryOrgId: this.supervisoryOrgId
    };
  }
};

// src/people-data/worker/types.ts
var PeopleDataWorkerAriResourceOwner = "people-data", PeopleDataWorkerAriResourceType = "worker";

// src/people-data/worker/manifest.ts
var peopleDataWorkerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataWorkerAriResourceOwner,
  resourceType: PeopleDataWorkerAriResourceType,
  resourceIdSlug: "{workerId}",
  resourceIdSegmentFormats: {
    workerId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/worker/index.ts
var PeopleDataWorkerAri = class _PeopleDataWorkerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workerId = opts.resourceIdSegmentValues.workerId;
  }
  get workerId() {
    return this._workerId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataWorkerAriStaticOpts.qualifier,
      platformQualifier: peopleDataWorkerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataWorkerAriStaticOpts.resourceOwner,
      resourceType: peopleDataWorkerAriStaticOpts.resourceType,
      resourceId: `${opts.workerId}`,
      resourceIdSegmentValues: {
        workerId: opts.workerId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataWorkerAriStaticOpts);
    return new _PeopleDataWorkerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataWorkerAriStaticOpts);
    return new _PeopleDataWorkerAri(opts);
  }
  getVariables() {
    return {
      workerId: this.workerId
    };
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PeopleDataCostCenterAri,
  PeopleDataJobProfileAri,
  PeopleDataPositionAri,
  PeopleDataSupervisoryOrgAri,
  PeopleDataWorkerAri
});
