// @forge/util/packages/ari/chunk-LVG3ZPQL.mjs
import {
  JiraSoftwareCustomFilterAriResourceOwner,
  JiraSoftwareCustomFilterAriResourceType
} from "./chunk-O6OVDRBO.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-software/custom-filter/manifest.ts
var jiraSoftwareCustomFilterAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: JiraSoftwareCustomFilterAriResourceOwner,
  resourceType: JiraSoftwareCustomFilterAriResourceType,
  resourceIdSlug: "activation/{activationId}/{boardId}/{customFilterId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    boardId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    customFilterId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-software/custom-filter/index.ts
var JiraSoftwareCustomFilterAri = class _JiraSoftwareCustomFilterAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._boardId = opts.resourceIdSegmentValues.boardId, this._customFilterId = opts.resourceIdSegmentValues.customFilterId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get boardId() {
    return this._boardId;
  }
  get customFilterId() {
    return this._customFilterId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraSoftwareCustomFilterAriStaticOpts.qualifier,
      platformQualifier: jiraSoftwareCustomFilterAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraSoftwareCustomFilterAriStaticOpts.resourceOwner,
      resourceType: jiraSoftwareCustomFilterAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.boardId}/${opts.customFilterId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        boardId: opts.boardId,
        customFilterId: opts.customFilterId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraSoftwareCustomFilterAriStaticOpts);
    return new _JiraSoftwareCustomFilterAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraSoftwareCustomFilterAriStaticOpts);
    return new _JiraSoftwareCustomFilterAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      boardId: this.boardId,
      customFilterId: this.customFilterId
    };
  }
};

export {
  JiraSoftwareCustomFilterAri
};
