// @forge/util/packages/ari/chunk-5SDHHCO3.mjs
import {
  PeoplePerftoolFeedbackAriResourceOwner,
  PeoplePerftoolFeedbackAriResourceType
} from "./chunk-BUB6ILHZ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/people-perftool/feedback/manifest.ts
var peoplePerftoolFeedbackAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeoplePerftoolFeedbackAriResourceOwner,
  resourceType: PeoplePerftoolFeedbackAriResourceType,
  resourceIdSlug: "{id}",
  resourceIdSegmentFormats: {
    id: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-perftool/feedback/index.ts
var PeoplePerftoolFeedbackAri = class _PeoplePerftoolFeedbackAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._id = opts.resourceIdSegmentValues.id;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peoplePerftoolFeedbackAriStaticOpts.qualifier,
      platformQualifier: peoplePerftoolFeedbackAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peoplePerftoolFeedbackAriStaticOpts.resourceOwner,
      resourceType: peoplePerftoolFeedbackAriStaticOpts.resourceType,
      resourceId: `${opts.id}`,
      resourceIdSegmentValues: {
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peoplePerftoolFeedbackAriStaticOpts);
    return new _PeoplePerftoolFeedbackAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peoplePerftoolFeedbackAriStaticOpts);
    return new _PeoplePerftoolFeedbackAri(opts);
  }
  getVariables() {
    return {
      id: this.id
    };
  }
};

export {
  PeoplePerftoolFeedbackAri
};
