// @forge/util/packages/ari/chunk-OF7MDY5D.mjs
import {
  BeaconRoleAriResourceOwner,
  BeaconRoleAriResourceType
} from "./chunk-YRF7LNG2.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/beacon/role/manifest.ts
var beaconRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: BeaconRoleAriResourceOwner,
  resourceType: BeaconRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:member|admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/beacon/role/index.ts
var BeaconRoleAri = class _BeaconRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: beaconRoleAriStaticOpts.qualifier,
      platformQualifier: beaconRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: beaconRoleAriStaticOpts.resourceOwner,
      resourceType: beaconRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, beaconRoleAriStaticOpts);
    return new _BeaconRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, beaconRoleAriStaticOpts);
    return new _BeaconRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  BeaconRoleAri
};
