// @forge/util/packages/ari/chunk-DOLO7K3M.mjs
import {
  GooglePresentationAriResourceOwner,
  GooglePresentationAriResourceType
} from "./chunk-5B6S6SWP.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/google/presentation/manifest.ts
var googlePresentationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: GooglePresentationAriResourceOwner,
  resourceType: GooglePresentationAriResourceType,
  resourceIdSlug: "{docId}",
  resourceIdSegmentFormats: {
    docId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/google/presentation/index.ts
var GooglePresentationAri = class _GooglePresentationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._docId = opts.resourceIdSegmentValues.docId;
  }
  get docId() {
    return this._docId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googlePresentationAriStaticOpts.qualifier,
      platformQualifier: googlePresentationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googlePresentationAriStaticOpts.resourceOwner,
      resourceType: googlePresentationAriStaticOpts.resourceType,
      resourceId: `${opts.docId}`,
      resourceIdSegmentValues: {
        docId: opts.docId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googlePresentationAriStaticOpts);
    return new _GooglePresentationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googlePresentationAriStaticOpts);
    return new _GooglePresentationAri(opts);
  }
  getVariables() {
    return {
      docId: this.docId
    };
  }
};

export {
  GooglePresentationAri
};
