// @forge/util/packages/ari/chunk-XJU7KB2W.mjs
import {
  AvocadoRoleAriResourceOwner,
  AvocadoRoleAriResourceType
} from "./chunk-DKX2RZ2Z.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/avocado/role/manifest.ts
var avocadoRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AvocadoRoleAriResourceOwner,
  resourceType: AvocadoRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:member|admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/avocado/role/index.ts
var AvocadoRoleAri = class _AvocadoRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avocadoRoleAriStaticOpts.qualifier,
      platformQualifier: avocadoRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: avocadoRoleAriStaticOpts.resourceOwner,
      resourceType: avocadoRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avocadoRoleAriStaticOpts);
    return new _AvocadoRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avocadoRoleAriStaticOpts);
    return new _AvocadoRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  AvocadoRoleAri
};
