// @forge/util/packages/ari/chunk-FLAS4OKN.mjs
import {
  GoogleFormAriResourceOwner,
  GoogleFormAriResourceType
} from "./chunk-X63IP3OI.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/google/form/manifest.ts
var googleFormAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: GoogleFormAriResourceOwner,
  resourceType: GoogleFormAriResourceType,
  resourceIdSlug: "{docId}",
  resourceIdSegmentFormats: {
    docId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/google/form/index.ts
var GoogleFormAri = class _GoogleFormAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._docId = opts.resourceIdSegmentValues.docId;
  }
  get docId() {
    return this._docId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleFormAriStaticOpts.qualifier,
      platformQualifier: googleFormAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleFormAriStaticOpts.resourceOwner,
      resourceType: googleFormAriStaticOpts.resourceType,
      resourceId: `${opts.docId}`,
      resourceIdSegmentValues: {
        docId: opts.docId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleFormAriStaticOpts);
    return new _GoogleFormAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleFormAriStaticOpts);
    return new _GoogleFormAri(opts);
  }
  getVariables() {
    return {
      docId: this.docId
    };
  }
};

export {
  GoogleFormAri
};
