// @forge/util/packages/ari/chunk-AJK4OU4N.mjs
import {
  BrieRoleAriResourceOwner,
  BrieRoleAriResourceType
} from "./chunk-S5IRDM4L.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/brie/role/manifest.ts
var brieRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: BrieRoleAriResourceOwner,
  resourceType: BrieRoleAriResourceType,
  resourceIdSlug: "scope/{roleTypeId}",
  resourceIdSegmentFormats: {
    roleTypeId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/brie/role/index.ts
var BrieRoleAri = class _BrieRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleTypeId = opts.resourceIdSegmentValues.roleTypeId;
  }
  get roleTypeId() {
    return this._roleTypeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: brieRoleAriStaticOpts.qualifier,
      platformQualifier: brieRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: brieRoleAriStaticOpts.resourceOwner,
      resourceType: brieRoleAriStaticOpts.resourceType,
      resourceId: `scope/${opts.roleTypeId}`,
      resourceIdSegmentValues: {
        roleTypeId: opts.roleTypeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, brieRoleAriStaticOpts);
    return new _BrieRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, brieRoleAriStaticOpts);
    return new _BrieRoleAri(opts);
  }
  getVariables() {
    return {
      roleTypeId: this.roleTypeId
    };
  }
};

export {
  BrieRoleAri
};
