// @forge/util/packages/ari/chunk-SUMH47BG.mjs
import {
  MakerSpaceGroupAriResourceOwner,
  MakerSpaceGroupAriResourceType
} from "./chunk-TI7RQAMT.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/maker-space/group/manifest.ts
var makerSpaceGroupAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: MakerSpaceGroupAriResourceOwner,
  resourceType: MakerSpaceGroupAriResourceType,
  resourceIdSlug: "{groupId}",
  resourceIdSegmentFormats: {
    groupId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/maker-space/group/index.ts
var MakerSpaceGroupAri = class _MakerSpaceGroupAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._groupId = opts.resourceIdSegmentValues.groupId;
  }
  get groupId() {
    return this._groupId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: makerSpaceGroupAriStaticOpts.qualifier,
      platformQualifier: makerSpaceGroupAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: makerSpaceGroupAriStaticOpts.resourceOwner,
      resourceType: makerSpaceGroupAriStaticOpts.resourceType,
      resourceId: `${opts.groupId}`,
      resourceIdSegmentValues: {
        groupId: opts.groupId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, makerSpaceGroupAriStaticOpts);
    return new _MakerSpaceGroupAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, makerSpaceGroupAriStaticOpts);
    return new _MakerSpaceGroupAri(opts);
  }
  getVariables() {
    return {
      groupId: this.groupId
    };
  }
};

export {
  MakerSpaceGroupAri
};
