// @forge/util/packages/ari/chunk-7PXRISLH.mjs
import {
  GraphEmployeeAriResourceOwner,
  GraphEmployeeAriResourceType
} from "./chunk-3SMMF4WR.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/graph/employee/manifest.ts
var graphEmployeeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: GraphEmployeeAriResourceOwner,
  resourceType: GraphEmployeeAriResourceType,
  resourceIdSlug: "activation/{activationId}/{employeeId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    employeeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/employee/index.ts
var GraphEmployeeAri = class _GraphEmployeeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._employeeId = opts.resourceIdSegmentValues.employeeId;
  }
  get activationId() {
    return this._activationId;
  }
  get employeeId() {
    return this._employeeId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphEmployeeAriStaticOpts.qualifier,
      platformQualifier: graphEmployeeAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphEmployeeAriStaticOpts.resourceOwner,
      resourceType: graphEmployeeAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.employeeId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        employeeId: opts.employeeId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphEmployeeAriStaticOpts);
    return new _GraphEmployeeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphEmployeeAriStaticOpts);
    return new _GraphEmployeeAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      employeeId: this.employeeId
    };
  }
};

export {
  GraphEmployeeAri
};
