// @forge/util/packages/ari/chunk-ETMJSA36.mjs
import {
  CommerceChangeAriResourceOwner,
  CommerceChangeAriResourceType
} from "./chunk-R3MD4VCL.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/change/manifest.ts
var commerceChangeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceChangeAriResourceOwner,
  resourceType: CommerceChangeAriResourceType,
  resourceIdSlug: "{changeUuid}",
  resourceIdSegmentFormats: {
    changeUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/change/index.ts
var CommerceChangeAri = class _CommerceChangeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._changeUuid = opts.resourceIdSegmentValues.changeUuid;
  }
  get changeUuid() {
    return this._changeUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceChangeAriStaticOpts.qualifier,
      platformQualifier: commerceChangeAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceChangeAriStaticOpts.resourceOwner,
      resourceType: commerceChangeAriStaticOpts.resourceType,
      resourceId: `${opts.changeUuid}`,
      resourceIdSegmentValues: {
        changeUuid: opts.changeUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceChangeAriStaticOpts);
    return new _CommerceChangeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceChangeAriStaticOpts);
    return new _CommerceChangeAri(opts);
  }
  getVariables() {
    return {
      changeUuid: this.changeUuid
    };
  }
};

export {
  CommerceChangeAri
};
