// @forge/util/packages/ari/chunk-7KC4WFNP.mjs
import {
  PostOfficeMessageTemplateAriResourceOwner,
  PostOfficeMessageTemplateAriResourceType
} from "./chunk-YT2S6YE6.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/post-office/message-template/manifest.ts
var postOfficeMessageTemplateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PostOfficeMessageTemplateAriResourceOwner,
  resourceType: PostOfficeMessageTemplateAriResourceType,
  resourceIdSlug: "{messageTemplateId}",
  resourceIdSegmentFormats: {
    messageTemplateId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/post-office/message-template/index.ts
var PostOfficeMessageTemplateAri = class _PostOfficeMessageTemplateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._messageTemplateId = opts.resourceIdSegmentValues.messageTemplateId;
  }
  get messageTemplateId() {
    return this._messageTemplateId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: postOfficeMessageTemplateAriStaticOpts.qualifier,
      platformQualifier: postOfficeMessageTemplateAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: postOfficeMessageTemplateAriStaticOpts.resourceOwner,
      resourceType: postOfficeMessageTemplateAriStaticOpts.resourceType,
      resourceId: `${opts.messageTemplateId}`,
      resourceIdSegmentValues: {
        messageTemplateId: opts.messageTemplateId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, postOfficeMessageTemplateAriStaticOpts);
    return new _PostOfficeMessageTemplateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, postOfficeMessageTemplateAriStaticOpts);
    return new _PostOfficeMessageTemplateAri(opts);
  }
  getVariables() {
    return {
      messageTemplateId: this.messageTemplateId
    };
  }
};

export {
  PostOfficeMessageTemplateAri
};
