// @forge/util/packages/ari/chunk-D7YGYOLP.mjs
import {
  UnifiedHelpWorkspaceAriResourceOwner,
  UnifiedHelpWorkspaceAriResourceType
} from "./chunk-GXNUL5AQ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/unified-help/workspace/manifest.ts
var unifiedHelpWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: UnifiedHelpWorkspaceAriResourceOwner,
  resourceType: UnifiedHelpWorkspaceAriResourceType,
  resourceIdSlug: "{activationId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-help/workspace/index.ts
var UnifiedHelpWorkspaceAri = class _UnifiedHelpWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId;
  }
  get activationId() {
    return this._activationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedHelpWorkspaceAriStaticOpts.qualifier,
      platformQualifier: unifiedHelpWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedHelpWorkspaceAriStaticOpts.resourceOwner,
      resourceType: unifiedHelpWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.activationId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedHelpWorkspaceAriStaticOpts);
    return new _UnifiedHelpWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedHelpWorkspaceAriStaticOpts);
    return new _UnifiedHelpWorkspaceAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId
    };
  }
};

export {
  UnifiedHelpWorkspaceAri
};
