// @forge/util/packages/ari/chunk-2ENFZU2S.mjs
import {
  AvocadoQuestionAriResourceOwner,
  AvocadoQuestionAriResourceType
} from "./chunk-URX7NO2Y.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/avocado/question/manifest.ts
var avocadoQuestionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: AvocadoQuestionAriResourceOwner,
  resourceType: AvocadoQuestionAriResourceType,
  resourceIdSlug: "activation/{workspaceId}/{questionId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    questionId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avocado/question/index.ts
var AvocadoQuestionAri = class _AvocadoQuestionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._questionId = opts.resourceIdSegmentValues.questionId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get questionId() {
    return this._questionId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avocadoQuestionAriStaticOpts.qualifier,
      platformQualifier: avocadoQuestionAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avocadoQuestionAriStaticOpts.resourceOwner,
      resourceType: avocadoQuestionAriStaticOpts.resourceType,
      resourceId: `activation/${opts.workspaceId}/${opts.questionId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        questionId: opts.questionId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avocadoQuestionAriStaticOpts);
    return new _AvocadoQuestionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avocadoQuestionAriStaticOpts);
    return new _AvocadoQuestionAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      questionId: this.questionId
    };
  }
};

export {
  AvocadoQuestionAri
};
