// @forge/util/packages/ari/chunk-LQSKKICR.mjs
import {
  AutomationRoleAriResourceOwner,
  AutomationRoleAriResourceType
} from "./chunk-QUWGD5ND.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/automation/role/manifest.ts
var automationRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AutomationRoleAriResourceOwner,
  resourceType: AutomationRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/automation/role/index.ts
var AutomationRoleAri = class _AutomationRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: automationRoleAriStaticOpts.qualifier,
      platformQualifier: automationRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: automationRoleAriStaticOpts.resourceOwner,
      resourceType: automationRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, automationRoleAriStaticOpts);
    return new _AutomationRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, automationRoleAriStaticOpts);
    return new _AutomationRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  AutomationRoleAri
};
