// @forge/util/packages/ari/chunk-IKDZRK4X.mjs
import {
  GraphVideoAriResourceOwner,
  GraphVideoAriResourceType
} from "./chunk-PDTRZUQJ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/graph/video/manifest.ts
var graphVideoAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: GraphVideoAriResourceOwner,
  resourceType: GraphVideoAriResourceType,
  resourceIdSlug: "activation/{activationId}/{entityUUID}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    entityUUID: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/video/index.ts
var GraphVideoAri = class _GraphVideoAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._entityUUID = opts.resourceIdSegmentValues.entityUUID;
  }
  get activationId() {
    return this._activationId;
  }
  get entityUUID() {
    return this._entityUUID;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphVideoAriStaticOpts.qualifier,
      platformQualifier: graphVideoAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphVideoAriStaticOpts.resourceOwner,
      resourceType: graphVideoAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.entityUUID}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        entityUUID: opts.entityUUID
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphVideoAriStaticOpts);
    return new _GraphVideoAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphVideoAriStaticOpts);
    return new _GraphVideoAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      entityUUID: this.entityUUID
    };
  }
};

export {
  GraphVideoAri
};
