// @forge/util/packages/ari/chunk-7MFBH35M.mjs
import {
  JiraSoftwareSwimlaneAriResourceOwner,
  JiraSoftwareSwimlaneAriResourceType
} from "./chunk-IATVHO4Z.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-software/swimlane/manifest.ts
var jiraSoftwareSwimlaneAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: JiraSoftwareSwimlaneAriResourceOwner,
  resourceType: JiraSoftwareSwimlaneAriResourceType,
  resourceIdSlug: "activation/{activationId}/{boardId}/{swimlaneId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    boardId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    swimlaneId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-software/swimlane/index.ts
var JiraSoftwareSwimlaneAri = class _JiraSoftwareSwimlaneAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._boardId = opts.resourceIdSegmentValues.boardId, this._swimlaneId = opts.resourceIdSegmentValues.swimlaneId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get boardId() {
    return this._boardId;
  }
  get swimlaneId() {
    return this._swimlaneId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraSoftwareSwimlaneAriStaticOpts.qualifier,
      platformQualifier: jiraSoftwareSwimlaneAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraSoftwareSwimlaneAriStaticOpts.resourceOwner,
      resourceType: jiraSoftwareSwimlaneAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.boardId}/${opts.swimlaneId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        boardId: opts.boardId,
        swimlaneId: opts.swimlaneId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraSoftwareSwimlaneAriStaticOpts);
    return new _JiraSoftwareSwimlaneAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraSoftwareSwimlaneAriStaticOpts);
    return new _JiraSoftwareSwimlaneAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      boardId: this.boardId,
      swimlaneId: this.swimlaneId
    };
  }
};

export {
  JiraSoftwareSwimlaneAri
};
