// @forge/util/packages/ari/chunk-MAOFP7FH.mjs
import {
  WorkdayCostCenterAriResourceOwner,
  WorkdayCostCenterAriResourceType
} from "./chunk-4UYKCLC2.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/workday/cost-center/manifest.ts
var workdayCostCenterAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayCostCenterAriResourceOwner,
  resourceType: WorkdayCostCenterAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{costCenterWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    costCenterWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/cost-center/index.ts
var WorkdayCostCenterAri = class _WorkdayCostCenterAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._costCenterWorkdayId = opts.resourceIdSegmentValues.costCenterWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get costCenterWorkdayId() {
    return this._costCenterWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayCostCenterAriStaticOpts.qualifier,
      platformQualifier: workdayCostCenterAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayCostCenterAriStaticOpts.resourceOwner,
      resourceType: workdayCostCenterAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.costCenterWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        costCenterWorkdayId: opts.costCenterWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayCostCenterAriStaticOpts);
    return new _WorkdayCostCenterAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayCostCenterAriStaticOpts);
    return new _WorkdayCostCenterAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      costCenterWorkdayId: this.costCenterWorkdayId
    };
  }
};

export {
  WorkdayCostCenterAri
};
