// @forge/util/packages/ari/chunk-JM7RU6RV.mjs
import {
  JiraSoftwareCardParentAriResourceOwner,
  JiraSoftwareCardParentAriResourceType
} from "./chunk-M7R3FF5D.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-software/card-parent/manifest.ts
var jiraSoftwareCardParentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: JiraSoftwareCardParentAriResourceOwner,
  resourceType: JiraSoftwareCardParentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{boardId}/{cardParentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    boardId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    cardParentId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-software/card-parent/index.ts
var JiraSoftwareCardParentAri = class _JiraSoftwareCardParentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._boardId = opts.resourceIdSegmentValues.boardId, this._cardParentId = opts.resourceIdSegmentValues.cardParentId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get boardId() {
    return this._boardId;
  }
  get cardParentId() {
    return this._cardParentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraSoftwareCardParentAriStaticOpts.qualifier,
      platformQualifier: jiraSoftwareCardParentAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraSoftwareCardParentAriStaticOpts.resourceOwner,
      resourceType: jiraSoftwareCardParentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.boardId}/${opts.cardParentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        boardId: opts.boardId,
        cardParentId: opts.cardParentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraSoftwareCardParentAriStaticOpts);
    return new _JiraSoftwareCardParentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraSoftwareCardParentAriStaticOpts);
    return new _JiraSoftwareCardParentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      boardId: this.boardId,
      cardParentId: this.cardParentId
    };
  }
};

export {
  JiraSoftwareCardParentAri
};
