// @forge/util/packages/ari/chunk-HTT3RY3Q.mjs
import {
  PeopleDataCostCenterAriResourceOwner,
  PeopleDataCostCenterAriResourceType
} from "./chunk-KBWC62ZT.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/people-data/cost-center/manifest.ts
var peopleDataCostCenterAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataCostCenterAriResourceOwner,
  resourceType: PeopleDataCostCenterAriResourceType,
  resourceIdSlug: "{costCenterId}",
  resourceIdSegmentFormats: {
    costCenterId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/cost-center/index.ts
var PeopleDataCostCenterAri = class _PeopleDataCostCenterAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._costCenterId = opts.resourceIdSegmentValues.costCenterId;
  }
  get costCenterId() {
    return this._costCenterId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataCostCenterAriStaticOpts.qualifier,
      platformQualifier: peopleDataCostCenterAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataCostCenterAriStaticOpts.resourceOwner,
      resourceType: peopleDataCostCenterAriStaticOpts.resourceType,
      resourceId: `${opts.costCenterId}`,
      resourceIdSegmentValues: {
        costCenterId: opts.costCenterId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataCostCenterAriStaticOpts);
    return new _PeopleDataCostCenterAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataCostCenterAriStaticOpts);
    return new _PeopleDataCostCenterAri(opts);
  }
  getVariables() {
    return {
      costCenterId: this.costCenterId
    };
  }
};

export {
  PeopleDataCostCenterAri
};
