// @forge/util/packages/ari/chunk-MR5RWBYA.mjs
import {
  AnalyticsUserAriResourceOwner,
  AnalyticsUserAriResourceType
} from "./chunk-77DTJ44V.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/analytics/user/manifest.ts
var analyticsUserAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AnalyticsUserAriResourceOwner,
  resourceType: AnalyticsUserAriResourceType,
  resourceIdSlug: "anonymous/{anonUserId}",
  resourceIdSegmentFormats: {
    anonUserId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/analytics/user/index.ts
var AnalyticsUserAri = class _AnalyticsUserAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._anonUserId = opts.resourceIdSegmentValues.anonUserId;
  }
  get anonUserId() {
    return this._anonUserId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: analyticsUserAriStaticOpts.qualifier,
      platformQualifier: analyticsUserAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: analyticsUserAriStaticOpts.resourceOwner,
      resourceType: analyticsUserAriStaticOpts.resourceType,
      resourceId: `anonymous/${opts.anonUserId}`,
      resourceIdSegmentValues: {
        anonUserId: opts.anonUserId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, analyticsUserAriStaticOpts);
    return new _AnalyticsUserAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, analyticsUserAriStaticOpts);
    return new _AnalyticsUserAri(opts);
  }
  getVariables() {
    return {
      anonUserId: this.anonUserId
    };
  }
};

export {
  AnalyticsUserAri
};
