// @forge/util/packages/ari/chunk-I6U47KDP.mjs
import {
  GraphProjectAriResourceOwner,
  GraphProjectAriResourceType
} from "./chunk-XTUR6LTI.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/graph/project/manifest.ts
var graphProjectAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: GraphProjectAriResourceOwner,
  resourceType: GraphProjectAriResourceType,
  resourceIdSlug: "activation/{activationId}/{projectId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    projectId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/project/index.ts
var GraphProjectAri = class _GraphProjectAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._projectId = opts.resourceIdSegmentValues.projectId;
  }
  get activationId() {
    return this._activationId;
  }
  get projectId() {
    return this._projectId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphProjectAriStaticOpts.qualifier,
      platformQualifier: graphProjectAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphProjectAriStaticOpts.resourceOwner,
      resourceType: graphProjectAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.projectId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        projectId: opts.projectId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphProjectAriStaticOpts);
    return new _GraphProjectAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphProjectAriStaticOpts);
    return new _GraphProjectAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      projectId: this.projectId
    };
  }
};

export {
  GraphProjectAri
};
