// @forge/util/packages/ari/chunk-3QGB5KNZ.mjs
import {
  ElementsCustomEmojiMetadataAriResourceOwner,
  ElementsCustomEmojiMetadataAriResourceType
} from "./chunk-INZ6GYXP.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/elements/custom-emoji-metadata/manifest.ts
var elementsCustomEmojiMetadataAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: ElementsCustomEmojiMetadataAriResourceOwner,
  resourceType: ElementsCustomEmojiMetadataAriResourceType,
  resourceIdSlug: "{customEmojiMetadataId}",
  resourceIdSegmentFormats: {
    customEmojiMetadataId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/elements/custom-emoji-metadata/index.ts
var ElementsCustomEmojiMetadataAri = class _ElementsCustomEmojiMetadataAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._customEmojiMetadataId = opts.resourceIdSegmentValues.customEmojiMetadataId;
  }
  get customEmojiMetadataId() {
    return this._customEmojiMetadataId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: elementsCustomEmojiMetadataAriStaticOpts.qualifier,
      platformQualifier: elementsCustomEmojiMetadataAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: elementsCustomEmojiMetadataAriStaticOpts.resourceOwner,
      resourceType: elementsCustomEmojiMetadataAriStaticOpts.resourceType,
      resourceId: `${opts.customEmojiMetadataId}`,
      resourceIdSegmentValues: {
        customEmojiMetadataId: opts.customEmojiMetadataId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, elementsCustomEmojiMetadataAriStaticOpts);
    return new _ElementsCustomEmojiMetadataAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, elementsCustomEmojiMetadataAriStaticOpts);
    return new _ElementsCustomEmojiMetadataAri(opts);
  }
  getVariables() {
    return {
      customEmojiMetadataId: this.customEmojiMetadataId
    };
  }
};

export {
  ElementsCustomEmojiMetadataAri
};
