// @forge/util/packages/ari/chunk-Q3J6TCM7.mjs
import {
  ElementsWorkspaceAriResourceOwner,
  ElementsWorkspaceAriResourceType
} from "./chunk-5TZRCHUN.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/elements/workspace/manifest.ts
var elementsWorkspaceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: ElementsWorkspaceAriResourceOwner,
  resourceType: ElementsWorkspaceAriResourceType,
  resourceIdSlug: "{workspaceId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/elements/workspace/index.ts
var ElementsWorkspaceAri = class _ElementsWorkspaceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: elementsWorkspaceAriStaticOpts.qualifier,
      platformQualifier: elementsWorkspaceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: elementsWorkspaceAriStaticOpts.resourceOwner,
      resourceType: elementsWorkspaceAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, elementsWorkspaceAriStaticOpts);
    return new _ElementsWorkspaceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, elementsWorkspaceAriStaticOpts);
    return new _ElementsWorkspaceAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId
    };
  }
};

export {
  ElementsWorkspaceAri
};
