// @forge/util/packages/ari/chunk-UUXWYD7A.mjs
import {
  AvpDatasourceAriResourceOwner,
  AvpDatasourceAriResourceType
} from "./chunk-MIBCZ475.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/avp/datasource/manifest.ts
var avpDatasourceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: AvpDatasourceAriResourceOwner,
  resourceType: AvpDatasourceAriResourceType,
  resourceIdSlug: "activation/{activationId}/{datasourceId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    datasourceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/datasource/index.ts
var AvpDatasourceAri = class _AvpDatasourceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._datasourceId = opts.resourceIdSegmentValues.datasourceId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get datasourceId() {
    return this._datasourceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpDatasourceAriStaticOpts.qualifier,
      platformQualifier: avpDatasourceAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: avpDatasourceAriStaticOpts.resourceOwner,
      resourceType: avpDatasourceAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.datasourceId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        datasourceId: opts.datasourceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpDatasourceAriStaticOpts);
    return new _AvpDatasourceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpDatasourceAriStaticOpts);
    return new _AvpDatasourceAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      datasourceId: this.datasourceId
    };
  }
};

export {
  AvpDatasourceAri
};
