// @forge/util/packages/ari/chunk-OSF4PED7.mjs
import {
  JiraAlignEpicAriResourceOwner,
  JiraAlignEpicAriResourceType
} from "./chunk-5DRIX3CB.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-align/epic/manifest.ts
var jiraAlignEpicAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$"),
  resourceOwner: JiraAlignEpicAriResourceOwner,
  resourceType: JiraAlignEpicAriResourceType,
  resourceIdSlug: "{epicId}",
  resourceIdSegmentFormats: {
    epicId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-align/epic/index.ts
var JiraAlignEpicAri = class _JiraAlignEpicAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._instanceId = opts.cloudId || "", this._epicId = opts.resourceIdSegmentValues.epicId;
  }
  get instanceId() {
    return this._instanceId;
  }
  get epicId() {
    return this._epicId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraAlignEpicAriStaticOpts.qualifier,
      platformQualifier: jiraAlignEpicAriStaticOpts.platformQualifier,
      cloudId: opts.instanceId,
      resourceOwner: jiraAlignEpicAriStaticOpts.resourceOwner,
      resourceType: jiraAlignEpicAriStaticOpts.resourceType,
      resourceId: `${opts.epicId}`,
      resourceIdSegmentValues: {
        epicId: opts.epicId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraAlignEpicAriStaticOpts);
    return new _JiraAlignEpicAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraAlignEpicAriStaticOpts);
    return new _JiraAlignEpicAri(opts);
  }
  getVariables() {
    return {
      instanceId: this.instanceId,
      epicId: this.epicId
    };
  }
};

export {
  JiraAlignEpicAri
};
