// @forge/util/packages/ari/chunk-7OY3QZNC.mjs
import {
  CommerceEntitlementAriResourceOwner,
  CommerceEntitlementAriResourceType
} from "./chunk-NINDB5F3.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/entitlement/manifest.ts
var commerceEntitlementAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceEntitlementAriResourceOwner,
  resourceType: CommerceEntitlementAriResourceType,
  resourceIdSlug: "{entitlementId}",
  resourceIdSegmentFormats: {
    entitlementId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/entitlement/index.ts
var CommerceEntitlementAri = class _CommerceEntitlementAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._entitlementId = opts.resourceIdSegmentValues.entitlementId;
  }
  get entitlementId() {
    return this._entitlementId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceEntitlementAriStaticOpts.qualifier,
      platformQualifier: commerceEntitlementAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceEntitlementAriStaticOpts.resourceOwner,
      resourceType: commerceEntitlementAriStaticOpts.resourceType,
      resourceId: `${opts.entitlementId}`,
      resourceIdSegmentValues: {
        entitlementId: opts.entitlementId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceEntitlementAriStaticOpts);
    return new _CommerceEntitlementAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceEntitlementAriStaticOpts);
    return new _CommerceEntitlementAri(opts);
  }
  getVariables() {
    return {
      entitlementId: this.entitlementId
    };
  }
};

export {
  CommerceEntitlementAri
};
