// @forge/util/packages/ari/chunk-LZFI6LCG.mjs
import {
  AvocadoUserAriResourceOwner,
  AvocadoUserAriResourceType
} from "./chunk-3W3PTAEC.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/avocado/user/manifest.ts
var avocadoUserAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AvocadoUserAriResourceOwner,
  resourceType: AvocadoUserAriResourceType,
  resourceIdSlug: "{userId}",
  resourceIdSegmentFormats: {
    userId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/avocado/user/index.ts
var AvocadoUserAri = class _AvocadoUserAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._userId = opts.resourceIdSegmentValues.userId;
  }
  get userId() {
    return this._userId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avocadoUserAriStaticOpts.qualifier,
      platformQualifier: avocadoUserAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: avocadoUserAriStaticOpts.resourceOwner,
      resourceType: avocadoUserAriStaticOpts.resourceType,
      resourceId: `${opts.userId}`,
      resourceIdSegmentValues: {
        userId: opts.userId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avocadoUserAriStaticOpts);
    return new _AvocadoUserAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avocadoUserAriStaticOpts);
    return new _AvocadoUserAri(opts);
  }
  getVariables() {
    return {
      userId: this.userId
    };
  }
};

export {
  AvocadoUserAri
};
