// @forge/util/packages/ari/chunk-C5ZTYUYL.mjs
import {
  CommerceRoleAriResourceOwner,
  CommerceRoleAriResourceType
} from "./chunk-T3FMJP7C.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/role/manifest.ts
var commerceRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceRoleAriResourceOwner,
  resourceType: CommerceRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/role/index.ts
var CommerceRoleAri = class _CommerceRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceRoleAriStaticOpts.qualifier,
      platformQualifier: commerceRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceRoleAriStaticOpts.resourceOwner,
      resourceType: commerceRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceRoleAriStaticOpts);
    return new _CommerceRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceRoleAriStaticOpts);
    return new _CommerceRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  CommerceRoleAri
};
