// @forge/util/packages/ari/chunk-5CSBVRMS.mjs
import {
  CompassDocumentationCategoryAriResourceOwner,
  CompassDocumentationCategoryAriResourceType
} from "./chunk-BCHZ2JY7.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/compass/documentation-category/manifest.ts
var compassDocumentationCategoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: CompassDocumentationCategoryAriResourceOwner,
  resourceType: CompassDocumentationCategoryAriResourceType,
  resourceIdSlug: "{workspaceId}/{documentationCategoryId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    documentationCategoryId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/compass/documentation-category/index.ts
var CompassDocumentationCategoryAri = class _CompassDocumentationCategoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._documentationCategoryId = opts.resourceIdSegmentValues.documentationCategoryId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get documentationCategoryId() {
    return this._documentationCategoryId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: compassDocumentationCategoryAriStaticOpts.qualifier,
      platformQualifier: compassDocumentationCategoryAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: compassDocumentationCategoryAriStaticOpts.resourceOwner,
      resourceType: compassDocumentationCategoryAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.documentationCategoryId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        documentationCategoryId: opts.documentationCategoryId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, compassDocumentationCategoryAriStaticOpts);
    return new _CompassDocumentationCategoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, compassDocumentationCategoryAriStaticOpts);
    return new _CompassDocumentationCategoryAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      documentationCategoryId: this.documentationCategoryId
    };
  }
};

export {
  CompassDocumentationCategoryAri
};
