// @forge/util/packages/ari/chunk-NHHUUBEC.mjs
import {
  PeopleDataJobProfileAriResourceOwner,
  PeopleDataJobProfileAriResourceType
} from "./chunk-MYD6A7PA.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/people-data/job-profile/manifest.ts
var peopleDataJobProfileAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataJobProfileAriResourceOwner,
  resourceType: PeopleDataJobProfileAriResourceType,
  resourceIdSlug: "{jobProfileId}",
  resourceIdSegmentFormats: {
    jobProfileId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/job-profile/index.ts
var PeopleDataJobProfileAri = class _PeopleDataJobProfileAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._jobProfileId = opts.resourceIdSegmentValues.jobProfileId;
  }
  get jobProfileId() {
    return this._jobProfileId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataJobProfileAriStaticOpts.qualifier,
      platformQualifier: peopleDataJobProfileAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataJobProfileAriStaticOpts.resourceOwner,
      resourceType: peopleDataJobProfileAriStaticOpts.resourceType,
      resourceId: `${opts.jobProfileId}`,
      resourceIdSegmentValues: {
        jobProfileId: opts.jobProfileId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataJobProfileAriStaticOpts);
    return new _PeopleDataJobProfileAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataJobProfileAriStaticOpts);
    return new _PeopleDataJobProfileAri(opts);
  }
  getVariables() {
    return {
      jobProfileId: this.jobProfileId
    };
  }
};

export {
  PeopleDataJobProfileAri
};
