// @forge/util/packages/ari/chunk-AFMK7OUM.mjs
import {
  CommerceCatalogAccountAriResourceOwner,
  CommerceCatalogAccountAriResourceType
} from "./chunk-3FQB6Q73.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/catalog-account/manifest.ts
var commerceCatalogAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceCatalogAccountAriResourceOwner,
  resourceType: CommerceCatalogAccountAriResourceType,
  resourceIdSlug: "{catalogAccountId}",
  resourceIdSegmentFormats: {
    catalogAccountId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/catalog-account/index.ts
var CommerceCatalogAccountAri = class _CommerceCatalogAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._catalogAccountId = opts.resourceIdSegmentValues.catalogAccountId;
  }
  get catalogAccountId() {
    return this._catalogAccountId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceCatalogAccountAriStaticOpts.qualifier,
      platformQualifier: commerceCatalogAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceCatalogAccountAriStaticOpts.resourceOwner,
      resourceType: commerceCatalogAccountAriStaticOpts.resourceType,
      resourceId: `${opts.catalogAccountId}`,
      resourceIdSegmentValues: {
        catalogAccountId: opts.catalogAccountId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceCatalogAccountAriStaticOpts);
    return new _CommerceCatalogAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceCatalogAccountAriStaticOpts);
    return new _CommerceCatalogAccountAri(opts);
  }
  getVariables() {
    return {
      catalogAccountId: this.catalogAccountId
    };
  }
};

export {
  CommerceCatalogAccountAri
};
