// @forge/util/packages/ari/chunk-XKLGW67G.mjs
import {
  SearchEventAriResourceOwner,
  SearchEventAriResourceType
} from "./chunk-MLTMULJ4.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/search/event/manifest.ts
var searchEventAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: SearchEventAriResourceOwner,
  resourceType: SearchEventAriResourceType,
  resourceIdSlug: "{eventType}/{eventId}",
  resourceIdSegmentFormats: {
    eventType: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    eventId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/search/event/index.ts
var SearchEventAri = class _SearchEventAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._eventType = opts.resourceIdSegmentValues.eventType, this._eventId = opts.resourceIdSegmentValues.eventId;
  }
  get eventType() {
    return this._eventType;
  }
  get eventId() {
    return this._eventId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: searchEventAriStaticOpts.qualifier,
      platformQualifier: searchEventAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: searchEventAriStaticOpts.resourceOwner,
      resourceType: searchEventAriStaticOpts.resourceType,
      resourceId: `${opts.eventType}/${opts.eventId}`,
      resourceIdSegmentValues: {
        eventType: opts.eventType,
        eventId: opts.eventId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, searchEventAriStaticOpts);
    return new _SearchEventAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, searchEventAriStaticOpts);
    return new _SearchEventAri(opts);
  }
  getVariables() {
    return {
      eventType: this.eventType,
      eventId: this.eventId
    };
  }
};

export {
  SearchEventAri
};
