// @forge/util/packages/ari/chunk-PDNQ6GYK.mjs
import {
  WorkdayPositionAriResourceOwner,
  WorkdayPositionAriResourceType
} from "./chunk-4PEBAWWW.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/workday/position/manifest.ts
var workdayPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayPositionAriResourceOwner,
  resourceType: WorkdayPositionAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{positionWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    positionWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/position/index.ts
var WorkdayPositionAri = class _WorkdayPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._positionWorkdayId = opts.resourceIdSegmentValues.positionWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get positionWorkdayId() {
    return this._positionWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayPositionAriStaticOpts.qualifier,
      platformQualifier: workdayPositionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayPositionAriStaticOpts.resourceOwner,
      resourceType: workdayPositionAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.positionWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        positionWorkdayId: opts.positionWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayPositionAriStaticOpts);
    return new _WorkdayPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayPositionAriStaticOpts);
    return new _WorkdayPositionAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      positionWorkdayId: this.positionWorkdayId
    };
  }
};

export {
  WorkdayPositionAri
};
