// @forge/util/packages/ari/chunk-CVVDKEA6.mjs
import {
  OauthClientAriResourceOwner,
  OauthClientAriResourceType
} from "./chunk-SLRY2ZFP.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/oauth/client/manifest.ts
var oauthClientAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: OauthClientAriResourceOwner,
  resourceType: OauthClientAriResourceType,
  resourceIdSlug: "{clientId}",
  resourceIdSegmentFormats: {
    clientId: /[0-9a-zA-Z-_]{5,50}/
    // eslint-disable-line no-useless-escape
  }
};

// src/oauth/client/index.ts
var OauthClientAri = class _OauthClientAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._clientId = opts.resourceIdSegmentValues.clientId;
  }
  get clientId() {
    return this._clientId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: oauthClientAriStaticOpts.qualifier,
      platformQualifier: oauthClientAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: oauthClientAriStaticOpts.resourceOwner,
      resourceType: oauthClientAriStaticOpts.resourceType,
      resourceId: `${opts.clientId}`,
      resourceIdSegmentValues: {
        clientId: opts.clientId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, oauthClientAriStaticOpts);
    return new _OauthClientAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, oauthClientAriStaticOpts);
    return new _OauthClientAri(opts);
  }
  getVariables() {
    return {
      clientId: this.clientId
    };
  }
};

export {
  OauthClientAri
};
