// @forge/util/packages/ari/chunk-5GSFSXKC.mjs
import {
  GraphEmployeeHistoryAriResourceOwner,
  GraphEmployeeHistoryAriResourceType
} from "./chunk-EDEP7UFY.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/graph/employee-history/manifest.ts
var graphEmployeeHistoryAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: GraphEmployeeHistoryAriResourceOwner,
  resourceType: GraphEmployeeHistoryAriResourceType,
  resourceIdSlug: "activation/{activationId}/{employeeId}/{updateSequenceNumber}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    employeeId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    updateSequenceNumber: /\d+/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/employee-history/index.ts
var GraphEmployeeHistoryAri = class _GraphEmployeeHistoryAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._employeeId = opts.resourceIdSegmentValues.employeeId, this._updateSequenceNumber = opts.resourceIdSegmentValues.updateSequenceNumber;
  }
  get activationId() {
    return this._activationId;
  }
  get employeeId() {
    return this._employeeId;
  }
  get updateSequenceNumber() {
    return this._updateSequenceNumber;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphEmployeeHistoryAriStaticOpts.qualifier,
      platformQualifier: graphEmployeeHistoryAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphEmployeeHistoryAriStaticOpts.resourceOwner,
      resourceType: graphEmployeeHistoryAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.employeeId}/${opts.updateSequenceNumber}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        employeeId: opts.employeeId,
        updateSequenceNumber: opts.updateSequenceNumber
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphEmployeeHistoryAriStaticOpts);
    return new _GraphEmployeeHistoryAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphEmployeeHistoryAriStaticOpts);
    return new _GraphEmployeeHistoryAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      employeeId: this.employeeId,
      updateSequenceNumber: this.updateSequenceNumber
    };
  }
};

export {
  GraphEmployeeHistoryAri
};
