// @forge/util/packages/ari/chunk-UHW4TW4X.mjs
import {
  CommerceInitiativeAriResourceOwner,
  CommerceInitiativeAriResourceType
} from "./chunk-NEKFBJUG.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/initiative/manifest.ts
var commerceInitiativeAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceInitiativeAriResourceOwner,
  resourceType: CommerceInitiativeAriResourceType,
  resourceIdSlug: "{initiativeUuid}",
  resourceIdSegmentFormats: {
    initiativeUuid: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/initiative/index.ts
var CommerceInitiativeAri = class _CommerceInitiativeAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._initiativeUuid = opts.resourceIdSegmentValues.initiativeUuid;
  }
  get initiativeUuid() {
    return this._initiativeUuid;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceInitiativeAriStaticOpts.qualifier,
      platformQualifier: commerceInitiativeAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceInitiativeAriStaticOpts.resourceOwner,
      resourceType: commerceInitiativeAriStaticOpts.resourceType,
      resourceId: `${opts.initiativeUuid}`,
      resourceIdSegmentValues: {
        initiativeUuid: opts.initiativeUuid
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceInitiativeAriStaticOpts);
    return new _CommerceInitiativeAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceInitiativeAriStaticOpts);
    return new _CommerceInitiativeAri(opts);
  }
  getVariables() {
    return {
      initiativeUuid: this.initiativeUuid
    };
  }
};

export {
  CommerceInitiativeAri
};
