// @forge/util/packages/ari/chunk-LM2WATIN.mjs
import {
  GraphRemoteLinkAriResourceOwner,
  GraphRemoteLinkAriResourceType
} from "./chunk-7CH7D4HJ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/graph/remote-link/manifest.ts
var graphRemoteLinkAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: GraphRemoteLinkAriResourceOwner,
  resourceType: GraphRemoteLinkAriResourceType,
  resourceIdSlug: "activation/{activationId}/{remoteLinkId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    remoteLinkId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/graph/remote-link/index.ts
var GraphRemoteLinkAri = class _GraphRemoteLinkAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._activationId = opts.resourceIdSegmentValues.activationId, this._remoteLinkId = opts.resourceIdSegmentValues.remoteLinkId;
  }
  get activationId() {
    return this._activationId;
  }
  get remoteLinkId() {
    return this._remoteLinkId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: graphRemoteLinkAriStaticOpts.qualifier,
      platformQualifier: graphRemoteLinkAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: graphRemoteLinkAriStaticOpts.resourceOwner,
      resourceType: graphRemoteLinkAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.remoteLinkId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        remoteLinkId: opts.remoteLinkId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, graphRemoteLinkAriStaticOpts);
    return new _GraphRemoteLinkAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, graphRemoteLinkAriStaticOpts);
    return new _GraphRemoteLinkAri(opts);
  }
  getVariables() {
    return {
      activationId: this.activationId,
      remoteLinkId: this.remoteLinkId
    };
  }
};

export {
  GraphRemoteLinkAri
};
