// @forge/util/packages/ari/chunk-23SWKVWV.mjs
import {
  AtlashubSiteAriResourceOwner,
  AtlashubSiteAriResourceType
} from "./chunk-SJUWQEXL.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/atlashub/site/manifest.ts
var atlashubSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AtlashubSiteAriResourceOwner,
  resourceType: AtlashubSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/atlashub/site/index.ts
var AtlashubSiteAri = class _AtlashubSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: atlashubSiteAriStaticOpts.qualifier,
      platformQualifier: atlashubSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: atlashubSiteAriStaticOpts.resourceOwner,
      resourceType: atlashubSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, atlashubSiteAriStaticOpts);
    return new _AtlashubSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, atlashubSiteAriStaticOpts);
    return new _AtlashubSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  AtlashubSiteAri
};
