// @forge/util/packages/ari/chunk-ILC7JDSD.mjs
import {
  CanvasBoardAriResourceOwner,
  CanvasBoardAriResourceType
} from "./chunk-R3VA6JMD.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/canvas/board/manifest.ts
var canvasBoardAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: CanvasBoardAriResourceOwner,
  resourceType: CanvasBoardAriResourceType,
  resourceIdSlug: "{boardId}",
  resourceIdSegmentFormats: {
    boardId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/canvas/board/index.ts
var CanvasBoardAri = class _CanvasBoardAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._boardId = opts.resourceIdSegmentValues.boardId;
  }
  get siteId() {
    return this._siteId;
  }
  get boardId() {
    return this._boardId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: canvasBoardAriStaticOpts.qualifier,
      platformQualifier: canvasBoardAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: canvasBoardAriStaticOpts.resourceOwner,
      resourceType: canvasBoardAriStaticOpts.resourceType,
      resourceId: `${opts.boardId}`,
      resourceIdSegmentValues: {
        boardId: opts.boardId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, canvasBoardAriStaticOpts);
    return new _CanvasBoardAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, canvasBoardAriStaticOpts);
    return new _CanvasBoardAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      boardId: this.boardId
    };
  }
};

export {
  CanvasBoardAri
};
