// @forge/util/packages/ari/chunk-33ZBDAZK.mjs
import {
  ConfluenceNoteAriResourceOwner,
  ConfluenceNoteAriResourceType
} from "./chunk-ZWIRILL2.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/confluence/note/manifest.ts
var confluenceNoteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: ConfluenceNoteAriResourceOwner,
  resourceType: ConfluenceNoteAriResourceType,
  resourceIdSlug: "activation/{activationId}/{noteId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    noteId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence/note/index.ts
var ConfluenceNoteAri = class _ConfluenceNoteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._noteId = opts.resourceIdSegmentValues.noteId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get noteId() {
    return this._noteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceNoteAriStaticOpts.qualifier,
      platformQualifier: confluenceNoteAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: confluenceNoteAriStaticOpts.resourceOwner,
      resourceType: confluenceNoteAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.noteId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        noteId: opts.noteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceNoteAriStaticOpts);
    return new _ConfluenceNoteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceNoteAriStaticOpts);
    return new _ConfluenceNoteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      noteId: this.noteId
    };
  }
};

export {
  ConfluenceNoteAri
};
