// @forge/util/packages/ari/chunk-OOWLL7WB.mjs
import {
  BrieSiteAriResourceOwner,
  BrieSiteAriResourceType
} from "./chunk-2CDIPTTF.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/brie/site/manifest.ts
var brieSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: BrieSiteAriResourceOwner,
  resourceType: BrieSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/brie/site/index.ts
var BrieSiteAri = class _BrieSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: brieSiteAriStaticOpts.qualifier,
      platformQualifier: brieSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: brieSiteAriStaticOpts.resourceOwner,
      resourceType: brieSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, brieSiteAriStaticOpts);
    return new _BrieSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, brieSiteAriStaticOpts);
    return new _BrieSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  BrieSiteAri
};
