// @forge/util/packages/ari/chunk-Y2S7AKDG.mjs
import {
  IdentityThirdPartyGroupAriResourceOwner,
  IdentityThirdPartyGroupAriResourceType
} from "./chunk-IMAEBUUM.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/identity/third-party-group/manifest.ts
var identityThirdPartyGroupAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: IdentityThirdPartyGroupAriResourceOwner,
  resourceType: IdentityThirdPartyGroupAriResourceType,
  resourceIdSlug: "{directoryId}/{groupId}",
  resourceIdSegmentFormats: {
    directoryId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    groupId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/identity/third-party-group/index.ts
var IdentityThirdPartyGroupAri = class _IdentityThirdPartyGroupAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._directoryId = opts.resourceIdSegmentValues.directoryId, this._groupId = opts.resourceIdSegmentValues.groupId;
  }
  get directoryId() {
    return this._directoryId;
  }
  get groupId() {
    return this._groupId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: identityThirdPartyGroupAriStaticOpts.qualifier,
      platformQualifier: identityThirdPartyGroupAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: identityThirdPartyGroupAriStaticOpts.resourceOwner,
      resourceType: identityThirdPartyGroupAriStaticOpts.resourceType,
      resourceId: `${opts.directoryId}/${opts.groupId}`,
      resourceIdSegmentValues: {
        directoryId: opts.directoryId,
        groupId: opts.groupId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, identityThirdPartyGroupAriStaticOpts);
    return new _IdentityThirdPartyGroupAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, identityThirdPartyGroupAriStaticOpts);
    return new _IdentityThirdPartyGroupAri(opts);
  }
  getVariables() {
    return {
      directoryId: this.directoryId,
      groupId: this.groupId
    };
  }
};

export {
  IdentityThirdPartyGroupAri
};
