// @forge/util/packages/ari/chunk-W5VFP5CH.mjs
import {
  PapiRoleAriResourceOwner,
  PapiRoleAriResourceType
} from "./chunk-NLJJ6GJ4.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/papi/role/manifest.ts
var papiRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PapiRoleAriResourceOwner,
  resourceType: PapiRoleAriResourceType,
  resourceIdSlug: "partner",
  resourceIdSegmentFormats: {}
};

// src/papi/role/index.ts
var PapiRoleAri = class _PapiRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: papiRoleAriStaticOpts.qualifier,
      platformQualifier: papiRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: papiRoleAriStaticOpts.resourceOwner,
      resourceType: papiRoleAriStaticOpts.resourceType,
      resourceId: "partner",
      resourceIdSegmentValues: {}
    }, ariOpts = AriParser.fromOpts(derivedOpts, papiRoleAriStaticOpts);
    return new _PapiRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, papiRoleAriStaticOpts);
    return new _PapiRoleAri(opts);
  }
  getVariables() {
    return {};
  }
};

export {
  PapiRoleAri
};
