// @forge/util/packages/ari/chunk-ANGBESZT.mjs
import {
  CommerceWorkbenchResourceAriResourceOwner,
  CommerceWorkbenchResourceAriResourceType
} from "./chunk-CVI37ASZ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/workbench-resource/manifest.ts
var commerceWorkbenchResourceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceWorkbenchResourceAriResourceOwner,
  resourceType: CommerceWorkbenchResourceAriResourceType,
  resourceIdSlug: "{resourceName}",
  resourceIdSegmentFormats: {
    resourceName: /(database)/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/workbench-resource/index.ts
var CommerceWorkbenchResourceAri = class _CommerceWorkbenchResourceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._resourceName = opts.resourceIdSegmentValues.resourceName;
  }
  get resourceName() {
    return this._resourceName;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceWorkbenchResourceAriStaticOpts.qualifier,
      platformQualifier: commerceWorkbenchResourceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceWorkbenchResourceAriStaticOpts.resourceOwner,
      resourceType: commerceWorkbenchResourceAriStaticOpts.resourceType,
      resourceId: `${opts.resourceName}`,
      resourceIdSegmentValues: {
        resourceName: opts.resourceName
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceWorkbenchResourceAriStaticOpts);
    return new _CommerceWorkbenchResourceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceWorkbenchResourceAriStaticOpts);
    return new _CommerceWorkbenchResourceAri(opts);
  }
  getVariables() {
    return {
      resourceName: this.resourceName
    };
  }
};

export {
  CommerceWorkbenchResourceAri
};
