// @forge/util/packages/ari/chunk-Q5ULMPIP.mjs
import {
  MercuryFocusAreaAriResourceOwner,
  MercuryFocusAreaAriResourceType
} from "./chunk-3SEGWH74.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/mercury/focus-area/manifest.ts
var mercuryFocusAreaAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: MercuryFocusAreaAriResourceOwner,
  resourceType: MercuryFocusAreaAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{focusAreaId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    focusAreaId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/mercury/focus-area/index.ts
var MercuryFocusAreaAri = class _MercuryFocusAreaAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._focusAreaId = opts.resourceIdSegmentValues.focusAreaId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get focusAreaId() {
    return this._focusAreaId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mercuryFocusAreaAriStaticOpts.qualifier,
      platformQualifier: mercuryFocusAreaAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: mercuryFocusAreaAriStaticOpts.resourceOwner,
      resourceType: mercuryFocusAreaAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.focusAreaId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        focusAreaId: opts.focusAreaId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mercuryFocusAreaAriStaticOpts);
    return new _MercuryFocusAreaAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mercuryFocusAreaAriStaticOpts);
    return new _MercuryFocusAreaAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      focusAreaId: this.focusAreaId
    };
  }
};

export {
  MercuryFocusAreaAri
};
