// @forge/util/packages/ari/chunk-AZXKOF4X.mjs
import {
  ConfluenceTaskAriResourceOwner,
  ConfluenceTaskAriResourceType
} from "./chunk-SHZM4TTX.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/confluence/task/manifest.ts
var confluenceTaskAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: ConfluenceTaskAriResourceOwner,
  resourceType: ConfluenceTaskAriResourceType,
  resourceIdSlug: "{taskId}",
  resourceIdSegmentFormats: {
    taskId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence/task/index.ts
var ConfluenceTaskAri = class _ConfluenceTaskAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._taskId = opts.resourceIdSegmentValues.taskId;
  }
  get siteId() {
    return this._siteId;
  }
  get taskId() {
    return this._taskId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceTaskAriStaticOpts.qualifier,
      platformQualifier: confluenceTaskAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: confluenceTaskAriStaticOpts.resourceOwner,
      resourceType: confluenceTaskAriStaticOpts.resourceType,
      resourceId: `${opts.taskId}`,
      resourceIdSegmentValues: {
        taskId: opts.taskId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceTaskAriStaticOpts);
    return new _ConfluenceTaskAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceTaskAriStaticOpts);
    return new _ConfluenceTaskAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      taskId: this.taskId
    };
  }
};

export {
  ConfluenceTaskAri
};
