// @forge/util/packages/ari/chunk-S5G53MVL.mjs
import {
  CommerceOrderAriResourceOwner,
  CommerceOrderAriResourceType
} from "./chunk-3AXUCZXV.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/order/manifest.ts
var commerceOrderAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: CommerceOrderAriResourceOwner,
  resourceType: CommerceOrderAriResourceType,
  resourceIdSlug: "{orderId}",
  resourceIdSegmentFormats: {
    orderId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/order/index.ts
var CommerceOrderAri = class _CommerceOrderAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._txaId = opts.cloudId || "", this._orderId = opts.resourceIdSegmentValues.orderId;
  }
  get txaId() {
    return this._txaId;
  }
  get orderId() {
    return this._orderId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceOrderAriStaticOpts.qualifier,
      platformQualifier: commerceOrderAriStaticOpts.platformQualifier,
      cloudId: opts.txaId,
      resourceOwner: commerceOrderAriStaticOpts.resourceOwner,
      resourceType: commerceOrderAriStaticOpts.resourceType,
      resourceId: `${opts.orderId}`,
      resourceIdSegmentValues: {
        orderId: opts.orderId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceOrderAriStaticOpts);
    return new _CommerceOrderAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceOrderAriStaticOpts);
    return new _CommerceOrderAri(opts);
  }
  getVariables() {
    return {
      txaId: this.txaId,
      orderId: this.orderId
    };
  }
};

export {
  CommerceOrderAri
};
