// @forge/util/packages/ari/chunk-YQO7Q5FB.mjs
import {
  ApprovalDeskDecisionTableAriResourceOwner,
  ApprovalDeskDecisionTableAriResourceType
} from "./chunk-GTNXGX3Z.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/approval-desk/decision-table/manifest.ts
var approvalDeskDecisionTableAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: ApprovalDeskDecisionTableAriResourceOwner,
  resourceType: ApprovalDeskDecisionTableAriResourceType,
  resourceIdSlug: "{workspaceId}/{workflowId}/{decisionTableId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    workflowId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    decisionTableId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/approval-desk/decision-table/index.ts
var ApprovalDeskDecisionTableAri = class _ApprovalDeskDecisionTableAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._workflowId = opts.resourceIdSegmentValues.workflowId, this._decisionTableId = opts.resourceIdSegmentValues.decisionTableId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get workflowId() {
    return this._workflowId;
  }
  get decisionTableId() {
    return this._decisionTableId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: approvalDeskDecisionTableAriStaticOpts.qualifier,
      platformQualifier: approvalDeskDecisionTableAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: approvalDeskDecisionTableAriStaticOpts.resourceOwner,
      resourceType: approvalDeskDecisionTableAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.workflowId}/${opts.decisionTableId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        workflowId: opts.workflowId,
        decisionTableId: opts.decisionTableId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, approvalDeskDecisionTableAriStaticOpts);
    return new _ApprovalDeskDecisionTableAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, approvalDeskDecisionTableAriStaticOpts);
    return new _ApprovalDeskDecisionTableAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      workflowId: this.workflowId,
      decisionTableId: this.decisionTableId
    };
  }
};

export {
  ApprovalDeskDecisionTableAri
};
