// @forge/util/packages/ari/chunk-3BGP6GF5.mjs
import {
  WorkdayJobProfileAriResourceOwner,
  WorkdayJobProfileAriResourceType
} from "./chunk-UO33O7I4.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/workday/job-profile/manifest.ts
var workdayJobProfileAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayJobProfileAriResourceOwner,
  resourceType: WorkdayJobProfileAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{jobProfileWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    jobProfileWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/job-profile/index.ts
var WorkdayJobProfileAri = class _WorkdayJobProfileAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._jobProfileWorkdayId = opts.resourceIdSegmentValues.jobProfileWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get jobProfileWorkdayId() {
    return this._jobProfileWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayJobProfileAriStaticOpts.qualifier,
      platformQualifier: workdayJobProfileAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayJobProfileAriStaticOpts.resourceOwner,
      resourceType: workdayJobProfileAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.jobProfileWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        jobProfileWorkdayId: opts.jobProfileWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayJobProfileAriStaticOpts);
    return new _WorkdayJobProfileAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayJobProfileAriStaticOpts);
    return new _WorkdayJobProfileAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      jobProfileWorkdayId: this.jobProfileWorkdayId
    };
  }
};

export {
  WorkdayJobProfileAri
};
