// @forge/util/packages/ari/chunk-TGPMZUC7.mjs
import {
  WorkdayStaffAriResourceOwner,
  WorkdayStaffAriResourceType
} from "./chunk-BXN33GTZ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/workday/staff/manifest.ts
var workdayStaffAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayStaffAriResourceOwner,
  resourceType: WorkdayStaffAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{staffId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    staffId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/staff/index.ts
var WorkdayStaffAri = class _WorkdayStaffAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._staffId = opts.resourceIdSegmentValues.staffId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get staffId() {
    return this._staffId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayStaffAriStaticOpts.qualifier,
      platformQualifier: workdayStaffAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayStaffAriStaticOpts.resourceOwner,
      resourceType: workdayStaffAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.staffId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        staffId: opts.staffId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayStaffAriStaticOpts);
    return new _WorkdayStaffAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayStaffAriStaticOpts);
    return new _WorkdayStaffAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      staffId: this.staffId
    };
  }
};

export {
  WorkdayStaffAri
};
