// @forge/util/packages/ari/chunk-SKAEK7YC.mjs
import {
  PlatformServicesStreamhubSchemaAriResourceOwner,
  PlatformServicesStreamhubSchemaAriResourceType
} from "./chunk-KM3YDOZJ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/platform-services/streamhub-schema/manifest.ts
var platformServicesStreamhubSchemaAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PlatformServicesStreamhubSchemaAriResourceOwner,
  resourceType: PlatformServicesStreamhubSchemaAriResourceType,
  resourceIdSlug: "{streamhubSchemaId}",
  resourceIdSegmentFormats: {
    streamhubSchemaId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/platform-services/streamhub-schema/index.ts
var PlatformServicesStreamhubSchemaAri = class _PlatformServicesStreamhubSchemaAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._streamhubSchemaId = opts.resourceIdSegmentValues.streamhubSchemaId;
  }
  get streamhubSchemaId() {
    return this._streamhubSchemaId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: platformServicesStreamhubSchemaAriStaticOpts.qualifier,
      platformQualifier: platformServicesStreamhubSchemaAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: platformServicesStreamhubSchemaAriStaticOpts.resourceOwner,
      resourceType: platformServicesStreamhubSchemaAriStaticOpts.resourceType,
      resourceId: `${opts.streamhubSchemaId}`,
      resourceIdSegmentValues: {
        streamhubSchemaId: opts.streamhubSchemaId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, platformServicesStreamhubSchemaAriStaticOpts);
    return new _PlatformServicesStreamhubSchemaAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, platformServicesStreamhubSchemaAriStaticOpts);
    return new _PlatformServicesStreamhubSchemaAri(opts);
  }
  getVariables() {
    return {
      streamhubSchemaId: this.streamhubSchemaId
    };
  }
};

export {
  PlatformServicesStreamhubSchemaAri
};
