// @forge/util/packages/ari/chunk-AKN2TPVJ.mjs
import {
  UnifiedPortableProfileUnifiedAccountAriResourceOwner,
  UnifiedPortableProfileUnifiedAccountAriResourceType
} from "./chunk-P5I3SEJO.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/unified-portable-profile/unified-account/manifest.ts
var unifiedPortableProfileUnifiedAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: UnifiedPortableProfileUnifiedAccountAriResourceOwner,
  resourceType: UnifiedPortableProfileUnifiedAccountAriResourceType,
  resourceIdSlug: "{linkedAccountInternalId}",
  resourceIdSegmentFormats: {
    linkedAccountInternalId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-portable-profile/unified-account/index.ts
var UnifiedPortableProfileUnifiedAccountAri = class _UnifiedPortableProfileUnifiedAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._linkedAccountInternalId = opts.resourceIdSegmentValues.linkedAccountInternalId;
  }
  get linkedAccountInternalId() {
    return this._linkedAccountInternalId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedPortableProfileUnifiedAccountAriStaticOpts.qualifier,
      platformQualifier: unifiedPortableProfileUnifiedAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedPortableProfileUnifiedAccountAriStaticOpts.resourceOwner,
      resourceType: unifiedPortableProfileUnifiedAccountAriStaticOpts.resourceType,
      resourceId: `${opts.linkedAccountInternalId}`,
      resourceIdSegmentValues: {
        linkedAccountInternalId: opts.linkedAccountInternalId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedPortableProfileUnifiedAccountAriStaticOpts);
    return new _UnifiedPortableProfileUnifiedAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedPortableProfileUnifiedAccountAriStaticOpts);
    return new _UnifiedPortableProfileUnifiedAccountAri(opts);
  }
  getVariables() {
    return {
      linkedAccountInternalId: this.linkedAccountInternalId
    };
  }
};

export {
  UnifiedPortableProfileUnifiedAccountAri
};
