// @forge/util/packages/ari/chunk-2DWFZ35E.mjs
import {
  MercuryProgramAriResourceOwner,
  MercuryProgramAriResourceType
} from "./chunk-YNUAOINR.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/mercury/program/manifest.ts
var mercuryProgramAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: MercuryProgramAriResourceOwner,
  resourceType: MercuryProgramAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{programId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    programId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/mercury/program/index.ts
var MercuryProgramAri = class _MercuryProgramAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._programId = opts.resourceIdSegmentValues.programId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get programId() {
    return this._programId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mercuryProgramAriStaticOpts.qualifier,
      platformQualifier: mercuryProgramAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: mercuryProgramAriStaticOpts.resourceOwner,
      resourceType: mercuryProgramAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.programId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        programId: opts.programId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mercuryProgramAriStaticOpts);
    return new _MercuryProgramAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mercuryProgramAriStaticOpts);
    return new _MercuryProgramAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      programId: this.programId
    };
  }
};

export {
  MercuryProgramAri
};
