// @forge/util/packages/ari/chunk-OZ67AOSA.mjs
import {
  UnifiedPortableProfileLinkedAccountAriResourceOwner,
  UnifiedPortableProfileLinkedAccountAriResourceType
} from "./chunk-GX4LHOQR.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/unified-portable-profile/linked-account/manifest.ts
var unifiedPortableProfileLinkedAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: UnifiedPortableProfileLinkedAccountAriResourceOwner,
  resourceType: UnifiedPortableProfileLinkedAccountAriResourceType,
  resourceIdSlug: "{linkedAccountInternalId}",
  resourceIdSegmentFormats: {
    linkedAccountInternalId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-portable-profile/linked-account/index.ts
var UnifiedPortableProfileLinkedAccountAri = class _UnifiedPortableProfileLinkedAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._linkedAccountInternalId = opts.resourceIdSegmentValues.linkedAccountInternalId;
  }
  get linkedAccountInternalId() {
    return this._linkedAccountInternalId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedPortableProfileLinkedAccountAriStaticOpts.qualifier,
      platformQualifier: unifiedPortableProfileLinkedAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedPortableProfileLinkedAccountAriStaticOpts.resourceOwner,
      resourceType: unifiedPortableProfileLinkedAccountAriStaticOpts.resourceType,
      resourceId: `${opts.linkedAccountInternalId}`,
      resourceIdSegmentValues: {
        linkedAccountInternalId: opts.linkedAccountInternalId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedPortableProfileLinkedAccountAriStaticOpts);
    return new _UnifiedPortableProfileLinkedAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedPortableProfileLinkedAccountAriStaticOpts);
    return new _UnifiedPortableProfileLinkedAccountAri(opts);
  }
  getVariables() {
    return {
      linkedAccountInternalId: this.linkedAccountInternalId
    };
  }
};

export {
  UnifiedPortableProfileLinkedAccountAri
};
