// @forge/util/packages/ari/chunk-5BCFBC4C.mjs
import {
  PeopleDataSupervisoryOrgAriResourceOwner,
  PeopleDataSupervisoryOrgAriResourceType
} from "./chunk-PA2MCDK3.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/people-data/supervisory-org/manifest.ts
var peopleDataSupervisoryOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataSupervisoryOrgAriResourceOwner,
  resourceType: PeopleDataSupervisoryOrgAriResourceType,
  resourceIdSlug: "{supervisoryOrgId}",
  resourceIdSegmentFormats: {
    supervisoryOrgId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/supervisory-org/index.ts
var PeopleDataSupervisoryOrgAri = class _PeopleDataSupervisoryOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._supervisoryOrgId = opts.resourceIdSegmentValues.supervisoryOrgId;
  }
  get supervisoryOrgId() {
    return this._supervisoryOrgId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataSupervisoryOrgAriStaticOpts.qualifier,
      platformQualifier: peopleDataSupervisoryOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataSupervisoryOrgAriStaticOpts.resourceOwner,
      resourceType: peopleDataSupervisoryOrgAriStaticOpts.resourceType,
      resourceId: `${opts.supervisoryOrgId}`,
      resourceIdSegmentValues: {
        supervisoryOrgId: opts.supervisoryOrgId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataSupervisoryOrgAriStaticOpts);
    return new _PeopleDataSupervisoryOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataSupervisoryOrgAriStaticOpts);
    return new _PeopleDataSupervisoryOrgAri(opts);
  }
  getVariables() {
    return {
      supervisoryOrgId: this.supervisoryOrgId
    };
  }
};

export {
  PeopleDataSupervisoryOrgAri
};
