// @forge/util/packages/ari/chunk-7OARX6B2.mjs
import {
  CommercePromotionAriResourceOwner,
  CommercePromotionAriResourceType
} from "./chunk-NDIFFXAM.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/promotion/manifest.ts
var commercePromotionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommercePromotionAriResourceOwner,
  resourceType: CommercePromotionAriResourceType,
  resourceIdSlug: "{promotionId}",
  resourceIdSegmentFormats: {
    promotionId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/promotion/index.ts
var CommercePromotionAri = class _CommercePromotionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._promotionId = opts.resourceIdSegmentValues.promotionId;
  }
  get promotionId() {
    return this._promotionId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commercePromotionAriStaticOpts.qualifier,
      platformQualifier: commercePromotionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commercePromotionAriStaticOpts.resourceOwner,
      resourceType: commercePromotionAriStaticOpts.resourceType,
      resourceId: `${opts.promotionId}`,
      resourceIdSegmentValues: {
        promotionId: opts.promotionId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commercePromotionAriStaticOpts);
    return new _CommercePromotionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commercePromotionAriStaticOpts);
    return new _CommercePromotionAri(opts);
  }
  getVariables() {
    return {
      promotionId: this.promotionId
    };
  }
};

export {
  CommercePromotionAri
};
