// @forge/util/packages/ari/chunk-SXJDQUBH.mjs
import {
  ConversationalHelpConversationAriResourceOwner,
  ConversationalHelpConversationAriResourceType
} from "./chunk-YWVV6JNH.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/conversational-help/conversation/manifest.ts
var conversationalHelpConversationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: ConversationalHelpConversationAriResourceOwner,
  resourceType: ConversationalHelpConversationAriResourceType,
  resourceIdSlug: "{conversationId}",
  resourceIdSegmentFormats: {
    conversationId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/conversational-help/conversation/index.ts
var ConversationalHelpConversationAri = class _ConversationalHelpConversationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._conversationId = opts.resourceIdSegmentValues.conversationId;
  }
  get siteId() {
    return this._siteId;
  }
  get conversationId() {
    return this._conversationId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: conversationalHelpConversationAriStaticOpts.qualifier,
      platformQualifier: conversationalHelpConversationAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: conversationalHelpConversationAriStaticOpts.resourceOwner,
      resourceType: conversationalHelpConversationAriStaticOpts.resourceType,
      resourceId: `${opts.conversationId}`,
      resourceIdSegmentValues: {
        conversationId: opts.conversationId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, conversationalHelpConversationAriStaticOpts);
    return new _ConversationalHelpConversationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, conversationalHelpConversationAriStaticOpts);
    return new _ConversationalHelpConversationAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      conversationId: this.conversationId
    };
  }
};

export {
  ConversationalHelpConversationAri
};
