// @forge/util/packages/ari/chunk-HLWWYVIZ.mjs
import {
  CommerceShipToPartyAriResourceOwner,
  CommerceShipToPartyAriResourceType
} from "./chunk-MR7K6E3K.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/ship-to-party/manifest.ts
var commerceShipToPartyAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: CommerceShipToPartyAriResourceOwner,
  resourceType: CommerceShipToPartyAriResourceType,
  resourceIdSlug: "{shipToPartyId}",
  resourceIdSegmentFormats: {
    shipToPartyId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/ship-to-party/index.ts
var CommerceShipToPartyAri = class _CommerceShipToPartyAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._txaId = opts.cloudId || "", this._shipToPartyId = opts.resourceIdSegmentValues.shipToPartyId;
  }
  get txaId() {
    return this._txaId;
  }
  get shipToPartyId() {
    return this._shipToPartyId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceShipToPartyAriStaticOpts.qualifier,
      platformQualifier: commerceShipToPartyAriStaticOpts.platformQualifier,
      cloudId: opts.txaId,
      resourceOwner: commerceShipToPartyAriStaticOpts.resourceOwner,
      resourceType: commerceShipToPartyAriStaticOpts.resourceType,
      resourceId: `${opts.shipToPartyId}`,
      resourceIdSegmentValues: {
        shipToPartyId: opts.shipToPartyId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceShipToPartyAriStaticOpts);
    return new _CommerceShipToPartyAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceShipToPartyAriStaticOpts);
    return new _CommerceShipToPartyAri(opts);
  }
  getVariables() {
    return {
      txaId: this.txaId,
      shipToPartyId: this.shipToPartyId
    };
  }
};

export {
  CommerceShipToPartyAri
};
