// @forge/util/packages/ari/chunk-UF3TVGFR.mjs
import {
  ApprovalDeskWorkflowAriResourceOwner,
  ApprovalDeskWorkflowAriResourceType
} from "./chunk-LIEBAJFZ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/approval-desk/workflow/manifest.ts
var approvalDeskWorkflowAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: ApprovalDeskWorkflowAriResourceOwner,
  resourceType: ApprovalDeskWorkflowAriResourceType,
  resourceIdSlug: "{workspaceId}/{workflowId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    workflowId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/approval-desk/workflow/index.ts
var ApprovalDeskWorkflowAri = class _ApprovalDeskWorkflowAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._workflowId = opts.resourceIdSegmentValues.workflowId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get workflowId() {
    return this._workflowId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: approvalDeskWorkflowAriStaticOpts.qualifier,
      platformQualifier: approvalDeskWorkflowAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: approvalDeskWorkflowAriStaticOpts.resourceOwner,
      resourceType: approvalDeskWorkflowAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.workflowId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        workflowId: opts.workflowId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, approvalDeskWorkflowAriStaticOpts);
    return new _ApprovalDeskWorkflowAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, approvalDeskWorkflowAriStaticOpts);
    return new _ApprovalDeskWorkflowAri(opts);
  }
  getVariables() {
    return {
      workspaceId: this.workspaceId,
      workflowId: this.workflowId
    };
  }
};

export {
  ApprovalDeskWorkflowAri
};
