// @forge/util/packages/ari/chunk-Q3BLDAKV.mjs
import {
  DevopsProviderAriResourceOwner,
  DevopsProviderAriResourceType
} from "./chunk-OILGX3NZ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/devops/provider/manifest.ts
var devopsProviderAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: DevopsProviderAriResourceOwner,
  resourceType: DevopsProviderAriResourceType,
  resourceIdSlug: "{providerNamespace}/{providerAppId}",
  resourceIdSegmentFormats: {
    providerNamespace: /(classic|forge|asap|oauth)/,
    // eslint-disable-line no-useless-escape
    providerAppId: /[a-zA-Z0-9-._]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/devops/provider/index.ts
var DevopsProviderAri = class _DevopsProviderAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._providerNamespace = opts.resourceIdSegmentValues.providerNamespace, this._providerAppId = opts.resourceIdSegmentValues.providerAppId;
  }
  get providerNamespace() {
    return this._providerNamespace;
  }
  get providerAppId() {
    return this._providerAppId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: devopsProviderAriStaticOpts.qualifier,
      platformQualifier: devopsProviderAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: devopsProviderAriStaticOpts.resourceOwner,
      resourceType: devopsProviderAriStaticOpts.resourceType,
      resourceId: `${opts.providerNamespace}/${opts.providerAppId}`,
      resourceIdSegmentValues: {
        providerNamespace: opts.providerNamespace,
        providerAppId: opts.providerAppId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, devopsProviderAriStaticOpts);
    return new _DevopsProviderAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, devopsProviderAriStaticOpts);
    return new _DevopsProviderAri(opts);
  }
  getVariables() {
    return {
      providerNamespace: this.providerNamespace,
      providerAppId: this.providerAppId
    };
  }
};

export {
  DevopsProviderAri
};
