// @forge/util/packages/ari/chunk-LSUGX3B2.mjs
import {
  ConfluenceAdminAnnouncementBannerAriResourceOwner,
  ConfluenceAdminAnnouncementBannerAriResourceType
} from "./chunk-IH6QYLXI.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/confluence/admin-announcement-banner/manifest.ts
var confluenceAdminAnnouncementBannerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: ConfluenceAdminAnnouncementBannerAriResourceOwner,
  resourceType: ConfluenceAdminAnnouncementBannerAriResourceType,
  resourceIdSlug: "activation/{activationId}/{bannerId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    bannerId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/confluence/admin-announcement-banner/index.ts
var ConfluenceAdminAnnouncementBannerAri = class _ConfluenceAdminAnnouncementBannerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._bannerId = opts.resourceIdSegmentValues.bannerId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get bannerId() {
    return this._bannerId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: confluenceAdminAnnouncementBannerAriStaticOpts.qualifier,
      platformQualifier: confluenceAdminAnnouncementBannerAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: confluenceAdminAnnouncementBannerAriStaticOpts.resourceOwner,
      resourceType: confluenceAdminAnnouncementBannerAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.bannerId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        bannerId: opts.bannerId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, confluenceAdminAnnouncementBannerAriStaticOpts);
    return new _ConfluenceAdminAnnouncementBannerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, confluenceAdminAnnouncementBannerAriStaticOpts);
    return new _ConfluenceAdminAnnouncementBannerAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      bannerId: this.bannerId
    };
  }
};

export {
  ConfluenceAdminAnnouncementBannerAri
};
