// @forge/util/packages/ari/chunk-EEVZ2T4N.mjs
import {
  CommerceEntitlementTemplateAriResourceOwner,
  CommerceEntitlementTemplateAriResourceType
} from "./chunk-32JDQQPU.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/entitlement-template/manifest.ts
var commerceEntitlementTemplateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceEntitlementTemplateAriResourceOwner,
  resourceType: CommerceEntitlementTemplateAriResourceType,
  resourceIdSlug: "{entitlementTemplateId}",
  resourceIdSegmentFormats: {
    entitlementTemplateId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/entitlement-template/index.ts
var CommerceEntitlementTemplateAri = class _CommerceEntitlementTemplateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._entitlementTemplateId = opts.resourceIdSegmentValues.entitlementTemplateId;
  }
  get entitlementTemplateId() {
    return this._entitlementTemplateId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceEntitlementTemplateAriStaticOpts.qualifier,
      platformQualifier: commerceEntitlementTemplateAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceEntitlementTemplateAriStaticOpts.resourceOwner,
      resourceType: commerceEntitlementTemplateAriStaticOpts.resourceType,
      resourceId: `${opts.entitlementTemplateId}`,
      resourceIdSegmentValues: {
        entitlementTemplateId: opts.entitlementTemplateId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceEntitlementTemplateAriStaticOpts);
    return new _CommerceEntitlementTemplateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceEntitlementTemplateAriStaticOpts);
    return new _CommerceEntitlementTemplateAri(opts);
  }
  getVariables() {
    return {
      entitlementTemplateId: this.entitlementTemplateId
    };
  }
};

export {
  CommerceEntitlementTemplateAri
};
