// @forge/util/packages/ari/chunk-HKHUOYMK.mjs
import {
  WorkdaySupervisoryOrgAriResourceOwner,
  WorkdaySupervisoryOrgAriResourceType
} from "./chunk-D7PCKJQ7.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/workday/supervisory-org/manifest.ts
var workdaySupervisoryOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdaySupervisoryOrgAriResourceOwner,
  resourceType: WorkdaySupervisoryOrgAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{supervisoryOrgWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    supervisoryOrgWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/supervisory-org/index.ts
var WorkdaySupervisoryOrgAri = class _WorkdaySupervisoryOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._supervisoryOrgWorkdayId = opts.resourceIdSegmentValues.supervisoryOrgWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get supervisoryOrgWorkdayId() {
    return this._supervisoryOrgWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdaySupervisoryOrgAriStaticOpts.qualifier,
      platformQualifier: workdaySupervisoryOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdaySupervisoryOrgAriStaticOpts.resourceOwner,
      resourceType: workdaySupervisoryOrgAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.supervisoryOrgWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        supervisoryOrgWorkdayId: opts.supervisoryOrgWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdaySupervisoryOrgAriStaticOpts);
    return new _WorkdaySupervisoryOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdaySupervisoryOrgAriStaticOpts);
    return new _WorkdaySupervisoryOrgAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      supervisoryOrgWorkdayId: this.supervisoryOrgWorkdayId
    };
  }
};

export {
  WorkdaySupervisoryOrgAri
};
