// @forge/util/packages/ari/chunk-W5AI5BPJ.mjs
import {
  RovoSiteAriResourceOwner,
  RovoSiteAriResourceType
} from "./chunk-JTG37E43.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/rovo/site/manifest.ts
var rovoSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: RovoSiteAriResourceOwner,
  resourceType: RovoSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/rovo/site/index.ts
var RovoSiteAri = class _RovoSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: rovoSiteAriStaticOpts.qualifier,
      platformQualifier: rovoSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: rovoSiteAriStaticOpts.resourceOwner,
      resourceType: rovoSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, rovoSiteAriStaticOpts);
    return new _RovoSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, rovoSiteAriStaticOpts);
    return new _RovoSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  RovoSiteAri
};
