// @forge/util/packages/ari/chunk-G3DXNZDR.mjs
import {
  JiraSoftwareRoleAriResourceOwner,
  JiraSoftwareRoleAriResourceType
} from "./chunk-SKSGTSPT.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-software/role/manifest.ts
var jiraSoftwareRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: JiraSoftwareRoleAriResourceOwner,
  resourceType: JiraSoftwareRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-software/role/index.ts
var JiraSoftwareRoleAri = class _JiraSoftwareRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraSoftwareRoleAriStaticOpts.qualifier,
      platformQualifier: jiraSoftwareRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraSoftwareRoleAriStaticOpts.resourceOwner,
      resourceType: jiraSoftwareRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraSoftwareRoleAriStaticOpts);
    return new _JiraSoftwareRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraSoftwareRoleAriStaticOpts);
    return new _JiraSoftwareRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  JiraSoftwareRoleAri
};
