// @forge/util/packages/ari/chunk-RWBTYLYR.mjs
import {
  CommerceProductAriResourceOwner,
  CommerceProductAriResourceType
} from "./chunk-7KRTMBHT.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/product/manifest.ts
var commerceProductAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceProductAriResourceOwner,
  resourceType: CommerceProductAriResourceType,
  resourceIdSlug: "{productId}",
  resourceIdSegmentFormats: {
    productId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/product/index.ts
var CommerceProductAri = class _CommerceProductAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._productId = opts.resourceIdSegmentValues.productId;
  }
  get productId() {
    return this._productId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceProductAriStaticOpts.qualifier,
      platformQualifier: commerceProductAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceProductAriStaticOpts.resourceOwner,
      resourceType: commerceProductAriStaticOpts.resourceType,
      resourceId: `${opts.productId}`,
      resourceIdSegmentValues: {
        productId: opts.productId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceProductAriStaticOpts);
    return new _CommerceProductAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceProductAriStaticOpts);
    return new _CommerceProductAri(opts);
  }
  getVariables() {
    return {
      productId: this.productId
    };
  }
};

export {
  CommerceProductAri
};
