// @forge/util/packages/ari/chunk-QSGN3HFR.mjs
import {
  MercuryProgramStatusUpdateAriResourceOwner,
  MercuryProgramStatusUpdateAriResourceType
} from "./chunk-DXGTR4X7.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/mercury/program-status-update/manifest.ts
var mercuryProgramStatusUpdateAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: MercuryProgramStatusUpdateAriResourceOwner,
  resourceType: MercuryProgramStatusUpdateAriResourceType,
  resourceIdSlug: "workspace/{workspaceId}/{programStatusUpdateId}",
  resourceIdSegmentFormats: {
    workspaceId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    programStatusUpdateId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/mercury/program-status-update/index.ts
var MercuryProgramStatusUpdateAri = class _MercuryProgramStatusUpdateAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._programStatusUpdateId = opts.resourceIdSegmentValues.programStatusUpdateId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get programStatusUpdateId() {
    return this._programStatusUpdateId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: mercuryProgramStatusUpdateAriStaticOpts.qualifier,
      platformQualifier: mercuryProgramStatusUpdateAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: mercuryProgramStatusUpdateAriStaticOpts.resourceOwner,
      resourceType: mercuryProgramStatusUpdateAriStaticOpts.resourceType,
      resourceId: `workspace/${opts.workspaceId}/${opts.programStatusUpdateId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        programStatusUpdateId: opts.programStatusUpdateId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, mercuryProgramStatusUpdateAriStaticOpts);
    return new _MercuryProgramStatusUpdateAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, mercuryProgramStatusUpdateAriStaticOpts);
    return new _MercuryProgramStatusUpdateAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      programStatusUpdateId: this.programStatusUpdateId
    };
  }
};

export {
  MercuryProgramStatusUpdateAri
};
