// @forge/util/packages/ari/chunk-MNXYH3HY.mjs
import {
  CommerceInvoiceGroupAriResourceOwner,
  CommerceInvoiceGroupAriResourceType
} from "./chunk-MX6GMSIL.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/invoice-group/manifest.ts
var commerceInvoiceGroupAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: CommerceInvoiceGroupAriResourceOwner,
  resourceType: CommerceInvoiceGroupAriResourceType,
  resourceIdSlug: "{invoiceGroupId}",
  resourceIdSegmentFormats: {
    invoiceGroupId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/invoice-group/index.ts
var CommerceInvoiceGroupAri = class _CommerceInvoiceGroupAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._txaId = opts.cloudId || "", this._invoiceGroupId = opts.resourceIdSegmentValues.invoiceGroupId;
  }
  get txaId() {
    return this._txaId;
  }
  get invoiceGroupId() {
    return this._invoiceGroupId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceInvoiceGroupAriStaticOpts.qualifier,
      platformQualifier: commerceInvoiceGroupAriStaticOpts.platformQualifier,
      cloudId: opts.txaId,
      resourceOwner: commerceInvoiceGroupAriStaticOpts.resourceOwner,
      resourceType: commerceInvoiceGroupAriStaticOpts.resourceType,
      resourceId: `${opts.invoiceGroupId}`,
      resourceIdSegmentValues: {
        invoiceGroupId: opts.invoiceGroupId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceInvoiceGroupAriStaticOpts);
    return new _CommerceInvoiceGroupAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceInvoiceGroupAriStaticOpts);
    return new _CommerceInvoiceGroupAri(opts);
  }
  getVariables() {
    return {
      txaId: this.txaId,
      invoiceGroupId: this.invoiceGroupId
    };
  }
};

export {
  CommerceInvoiceGroupAri
};
