// @forge/util/packages/ari/chunk-QOS4FHIS.mjs
import {
  PeopleDataPositionAriResourceOwner,
  PeopleDataPositionAriResourceType
} from "./chunk-GZISAL55.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/people-data/position/manifest.ts
var peopleDataPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeopleDataPositionAriResourceOwner,
  resourceType: PeopleDataPositionAriResourceType,
  resourceIdSlug: "{positionId}",
  resourceIdSegmentFormats: {
    positionId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-data/position/index.ts
var PeopleDataPositionAri = class _PeopleDataPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._positionId = opts.resourceIdSegmentValues.positionId;
  }
  get positionId() {
    return this._positionId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peopleDataPositionAriStaticOpts.qualifier,
      platformQualifier: peopleDataPositionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peopleDataPositionAriStaticOpts.resourceOwner,
      resourceType: peopleDataPositionAriStaticOpts.resourceType,
      resourceId: `${opts.positionId}`,
      resourceIdSegmentValues: {
        positionId: opts.positionId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peopleDataPositionAriStaticOpts);
    return new _PeopleDataPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peopleDataPositionAriStaticOpts);
    return new _PeopleDataPositionAri(opts);
  }
  getVariables() {
    return {
      positionId: this.positionId
    };
  }
};

export {
  PeopleDataPositionAri
};
