// @forge/util/packages/ari/chunk-EYTGQ6P4.mjs
import {
  UnifiedPortableProfileParentAccountAriResourceOwner,
  UnifiedPortableProfileParentAccountAriResourceType
} from "./chunk-PFRWENB5.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/unified-portable-profile/parent-account/manifest.ts
var unifiedPortableProfileParentAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: UnifiedPortableProfileParentAccountAriResourceOwner,
  resourceType: UnifiedPortableProfileParentAccountAriResourceType,
  resourceIdSlug: "{parentAccountInternalId}",
  resourceIdSegmentFormats: {
    parentAccountInternalId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-portable-profile/parent-account/index.ts
var UnifiedPortableProfileParentAccountAri = class _UnifiedPortableProfileParentAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._parentAccountInternalId = opts.resourceIdSegmentValues.parentAccountInternalId;
  }
  get parentAccountInternalId() {
    return this._parentAccountInternalId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedPortableProfileParentAccountAriStaticOpts.qualifier,
      platformQualifier: unifiedPortableProfileParentAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedPortableProfileParentAccountAriStaticOpts.resourceOwner,
      resourceType: unifiedPortableProfileParentAccountAriStaticOpts.resourceType,
      resourceId: `${opts.parentAccountInternalId}`,
      resourceIdSegmentValues: {
        parentAccountInternalId: opts.parentAccountInternalId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedPortableProfileParentAccountAriStaticOpts);
    return new _UnifiedPortableProfileParentAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedPortableProfileParentAccountAriStaticOpts);
    return new _UnifiedPortableProfileParentAccountAri(opts);
  }
  getVariables() {
    return {
      parentAccountInternalId: this.parentAccountInternalId
    };
  }
};

export {
  UnifiedPortableProfileParentAccountAri
};
