// @forge/util/packages/ari/chunk-H4FPMCLZ.mjs
import {
  IdentitySiteAriResourceOwner,
  IdentitySiteAriResourceType
} from "./chunk-CXXH2SWL.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/identity/site/manifest.ts
var identitySiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: IdentitySiteAriResourceOwner,
  resourceType: IdentitySiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/identity/site/index.ts
var IdentitySiteAri = class _IdentitySiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: identitySiteAriStaticOpts.qualifier,
      platformQualifier: identitySiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: identitySiteAriStaticOpts.resourceOwner,
      resourceType: identitySiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, identitySiteAriStaticOpts);
    return new _IdentitySiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, identitySiteAriStaticOpts);
    return new _IdentitySiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  IdentitySiteAri
};
