// @forge/util/packages/ari/chunk-TZ7NSRSU.mjs
import {
  StatuspageRoleAriResourceOwner,
  StatuspageRoleAriResourceType
} from "./chunk-BT2GYN5W.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/statuspage/role/manifest.ts
var statuspageRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: StatuspageRoleAriResourceOwner,
  resourceType: StatuspageRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/statuspage/role/index.ts
var StatuspageRoleAri = class _StatuspageRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: statuspageRoleAriStaticOpts.qualifier,
      platformQualifier: statuspageRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: statuspageRoleAriStaticOpts.resourceOwner,
      resourceType: statuspageRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, statuspageRoleAriStaticOpts);
    return new _StatuspageRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, statuspageRoleAriStaticOpts);
    return new _StatuspageRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  StatuspageRoleAri
};
