// @forge/util/packages/ari/chunk-LA3VCXUE.mjs
import {
  JiraCoreSiteAriResourceOwner,
  JiraCoreSiteAriResourceType
} from "./chunk-BYLK2PHJ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-core/site/manifest.ts
var jiraCoreSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: JiraCoreSiteAriResourceOwner,
  resourceType: JiraCoreSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-core/site/index.ts
var JiraCoreSiteAri = class _JiraCoreSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraCoreSiteAriStaticOpts.qualifier,
      platformQualifier: jiraCoreSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraCoreSiteAriStaticOpts.resourceOwner,
      resourceType: jiraCoreSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraCoreSiteAriStaticOpts);
    return new _JiraCoreSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraCoreSiteAriStaticOpts);
    return new _JiraCoreSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  JiraCoreSiteAri
};
