// @forge/util/packages/ari/chunk-C5TKHBWJ.mjs
import {
  CommerceOfferingAriResourceOwner,
  CommerceOfferingAriResourceType
} from "./chunk-ANF2XXE7.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/commerce/offering/manifest.ts
var commerceOfferingAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceOfferingAriResourceOwner,
  resourceType: CommerceOfferingAriResourceType,
  resourceIdSlug: "{offeringId}",
  resourceIdSegmentFormats: {
    offeringId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/offering/index.ts
var CommerceOfferingAri = class _CommerceOfferingAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._offeringId = opts.resourceIdSegmentValues.offeringId;
  }
  get offeringId() {
    return this._offeringId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceOfferingAriStaticOpts.qualifier,
      platformQualifier: commerceOfferingAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceOfferingAriStaticOpts.resourceOwner,
      resourceType: commerceOfferingAriStaticOpts.resourceType,
      resourceId: `${opts.offeringId}`,
      resourceIdSegmentValues: {
        offeringId: opts.offeringId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceOfferingAriStaticOpts);
    return new _CommerceOfferingAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceOfferingAriStaticOpts);
    return new _CommerceOfferingAri(opts);
  }
  getVariables() {
    return {
      offeringId: this.offeringId
    };
  }
};

export {
  CommerceOfferingAri
};
