// @forge/util/packages/ari/chunk-UG43ZMVT.mjs
import {
  BeaconSiteAriResourceOwner,
  BeaconSiteAriResourceType
} from "./chunk-JQQF6HSU.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/beacon/site/manifest.ts
var beaconSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: BeaconSiteAriResourceOwner,
  resourceType: BeaconSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/beacon/site/index.ts
var BeaconSiteAri = class _BeaconSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: beaconSiteAriStaticOpts.qualifier,
      platformQualifier: beaconSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: beaconSiteAriStaticOpts.resourceOwner,
      resourceType: beaconSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, beaconSiteAriStaticOpts);
    return new _BeaconSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, beaconSiteAriStaticOpts);
    return new _BeaconSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  BeaconSiteAri
};
