// @forge/util/packages/ari/chunk-KOYEMW72.mjs
import {
  JiraSoftwareCardAriResourceOwner,
  JiraSoftwareCardAriResourceType
} from "./chunk-6WK3XU6H.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-software/card/manifest.ts
var jiraSoftwareCardAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: JiraSoftwareCardAriResourceOwner,
  resourceType: JiraSoftwareCardAriResourceType,
  resourceIdSlug: "activation/{activationId}/{boardId}/{cardId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    boardId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    cardId: /[0-9]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-software/card/index.ts
var JiraSoftwareCardAri = class _JiraSoftwareCardAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._boardId = opts.resourceIdSegmentValues.boardId, this._cardId = opts.resourceIdSegmentValues.cardId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get boardId() {
    return this._boardId;
  }
  get cardId() {
    return this._cardId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraSoftwareCardAriStaticOpts.qualifier,
      platformQualifier: jiraSoftwareCardAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraSoftwareCardAriStaticOpts.resourceOwner,
      resourceType: jiraSoftwareCardAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.boardId}/${opts.cardId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        boardId: opts.boardId,
        cardId: opts.cardId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraSoftwareCardAriStaticOpts);
    return new _JiraSoftwareCardAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraSoftwareCardAriStaticOpts);
    return new _JiraSoftwareCardAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      boardId: this.boardId,
      cardId: this.cardId
    };
  }
};

export {
  JiraSoftwareCardAri
};
