// @forge/util/packages/ari/chunk-LGFWMBXU.mjs
import {
  IdentityOauthClientAriResourceOwner,
  IdentityOauthClientAriResourceType
} from "./chunk-UTUDGZLV.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/identity/oauthClient/manifest.ts
var identityOauthClientAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: IdentityOauthClientAriResourceOwner,
  resourceType: IdentityOauthClientAriResourceType,
  resourceIdSlug: "{clientId}",
  resourceIdSegmentFormats: {
    clientId: /[0-9a-zA-Z-_]{5,50}/
    // eslint-disable-line no-useless-escape
  }
};

// src/identity/oauthClient/index.ts
var IdentityOauthClientAri = class _IdentityOauthClientAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._clientId = opts.resourceIdSegmentValues.clientId;
  }
  get clientId() {
    return this._clientId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: identityOauthClientAriStaticOpts.qualifier,
      platformQualifier: identityOauthClientAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: identityOauthClientAriStaticOpts.resourceOwner,
      resourceType: identityOauthClientAriStaticOpts.resourceType,
      resourceId: `${opts.clientId}`,
      resourceIdSegmentValues: {
        clientId: opts.clientId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, identityOauthClientAriStaticOpts);
    return new _IdentityOauthClientAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, identityOauthClientAriStaticOpts);
    return new _IdentityOauthClientAri(opts);
  }
  getVariables() {
    return {
      clientId: this.clientId
    };
  }
};

export {
  IdentityOauthClientAri
};
