// @forge/util/packages/ari/chunk-PWZ2KYR5.mjs
import {
  CompassDocumentAriResourceOwner,
  CompassDocumentAriResourceType
} from "./chunk-TZHADK5A.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/compass/document/manifest.ts
var compassDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: CompassDocumentAriResourceOwner,
  resourceType: CompassDocumentAriResourceType,
  resourceIdSlug: "{workspaceId}/{documentId}",
  resourceIdSegmentFormats: {
    workspaceId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/,
    // eslint-disable-line no-useless-escape
    documentId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/compass/document/index.ts
var CompassDocumentAri = class _CompassDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._workspaceId = opts.resourceIdSegmentValues.workspaceId, this._documentId = opts.resourceIdSegmentValues.documentId;
  }
  get siteId() {
    return this._siteId;
  }
  get workspaceId() {
    return this._workspaceId;
  }
  get documentId() {
    return this._documentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: compassDocumentAriStaticOpts.qualifier,
      platformQualifier: compassDocumentAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: compassDocumentAriStaticOpts.resourceOwner,
      resourceType: compassDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.workspaceId}/${opts.documentId}`,
      resourceIdSegmentValues: {
        workspaceId: opts.workspaceId,
        documentId: opts.documentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, compassDocumentAriStaticOpts);
    return new _CompassDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, compassDocumentAriStaticOpts);
    return new _CompassDocumentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      workspaceId: this.workspaceId,
      documentId: this.documentId
    };
  }
};

export {
  CompassDocumentAri
};
