// @forge/util/packages/ari/chunk-A7WVIOCC.mjs
import {
  AvpRoleAriResourceOwner,
  AvpRoleAriResourceType
} from "./chunk-SZNOZQ4T.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/avp/role/manifest.ts
var avpRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AvpRoleAriResourceOwner,
  resourceType: AvpRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/role/index.ts
var AvpRoleAri = class _AvpRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpRoleAriStaticOpts.qualifier,
      platformQualifier: avpRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: avpRoleAriStaticOpts.resourceOwner,
      resourceType: avpRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpRoleAriStaticOpts);
    return new _AvpRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpRoleAriStaticOpts);
    return new _AvpRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  AvpRoleAri
};
