// @forge/util/packages/ari/chunk-MAGIGK4W.mjs
import {
  PassionfruitUserAriResourceOwner,
  PassionfruitUserAriResourceType
} from "./chunk-Q2W2OS63.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/passionfruit/user/manifest.ts
var passionfruitUserAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PassionfruitUserAriResourceOwner,
  resourceType: PassionfruitUserAriResourceType,
  resourceIdSlug: "{userId}",
  resourceIdSegmentFormats: {
    userId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/passionfruit/user/index.ts
var PassionfruitUserAri = class _PassionfruitUserAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._userId = opts.resourceIdSegmentValues.userId;
  }
  get userId() {
    return this._userId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: passionfruitUserAriStaticOpts.qualifier,
      platformQualifier: passionfruitUserAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: passionfruitUserAriStaticOpts.resourceOwner,
      resourceType: passionfruitUserAriStaticOpts.resourceType,
      resourceId: `${opts.userId}`,
      resourceIdSegmentValues: {
        userId: opts.userId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, passionfruitUserAriStaticOpts);
    return new _PassionfruitUserAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, passionfruitUserAriStaticOpts);
    return new _PassionfruitUserAri(opts);
  }
  getVariables() {
    return {
      userId: this.userId
    };
  }
};

export {
  PassionfruitUserAri
};
