// @forge/util/packages/ari/chunk-F5HMBVP2.mjs
import {
  AvpSiteAriResourceOwner,
  AvpSiteAriResourceType
} from "./chunk-GFZFXLJN.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/avp/site/manifest.ts
var avpSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: AvpSiteAriResourceOwner,
  resourceType: AvpSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/avp/site/index.ts
var AvpSiteAri = class _AvpSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: avpSiteAriStaticOpts.qualifier,
      platformQualifier: avpSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: avpSiteAriStaticOpts.resourceOwner,
      resourceType: avpSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, avpSiteAriStaticOpts);
    return new _AvpSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, avpSiteAriStaticOpts);
    return new _AvpSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  AvpSiteAri
};
