// @forge/util/packages/ari/chunk-2EBCBJ45.mjs
import {
  StatuspageSiteAriResourceOwner,
  StatuspageSiteAriResourceType
} from "./chunk-6IR7P2SS.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/statuspage/site/manifest.ts
var statuspageSiteAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: StatuspageSiteAriResourceOwner,
  resourceType: StatuspageSiteAriResourceType,
  resourceIdSlug: "{siteId}",
  resourceIdSegmentFormats: {
    siteId: /[a-zA-Z0-9\-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/statuspage/site/index.ts
var StatuspageSiteAri = class _StatuspageSiteAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.resourceIdSegmentValues.siteId;
  }
  get siteId() {
    return this._siteId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: statuspageSiteAriStaticOpts.qualifier,
      platformQualifier: statuspageSiteAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: statuspageSiteAriStaticOpts.resourceOwner,
      resourceType: statuspageSiteAriStaticOpts.resourceType,
      resourceId: `${opts.siteId}`,
      resourceIdSegmentValues: {
        siteId: opts.siteId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, statuspageSiteAriStaticOpts);
    return new _StatuspageSiteAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, statuspageSiteAriStaticOpts);
    return new _StatuspageSiteAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId
    };
  }
};

export {
  StatuspageSiteAri
};
