// @forge/util/packages/ari/chunk-HPL44SUH.mjs
import {
  PeoplePerftoolCalibrationAriResourceOwner,
  PeoplePerftoolCalibrationAriResourceType
} from "./chunk-YFDO5OIQ.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/people-perftool/calibration/manifest.ts
var peoplePerftoolCalibrationAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PeoplePerftoolCalibrationAriResourceOwner,
  resourceType: PeoplePerftoolCalibrationAriResourceType,
  resourceIdSlug: "{id}",
  resourceIdSegmentFormats: {
    id: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/people-perftool/calibration/index.ts
var PeoplePerftoolCalibrationAri = class _PeoplePerftoolCalibrationAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._id = opts.resourceIdSegmentValues.id;
  }
  get id() {
    return this._id;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: peoplePerftoolCalibrationAriStaticOpts.qualifier,
      platformQualifier: peoplePerftoolCalibrationAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: peoplePerftoolCalibrationAriStaticOpts.resourceOwner,
      resourceType: peoplePerftoolCalibrationAriStaticOpts.resourceType,
      resourceId: `${opts.id}`,
      resourceIdSegmentValues: {
        id: opts.id
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, peoplePerftoolCalibrationAriStaticOpts);
    return new _PeoplePerftoolCalibrationAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, peoplePerftoolCalibrationAriStaticOpts);
    return new _PeoplePerftoolCalibrationAri(opts);
  }
  getVariables() {
    return {
      id: this.id
    };
  }
};

export {
  PeoplePerftoolCalibrationAri
};
