// @forge/util/packages/ari/chunk-O5Z3IKRI.mjs
import {
  UnifiedPortableProfileUnifiedProfileAriResourceOwner,
  UnifiedPortableProfileUnifiedProfileAriResourceType
} from "./chunk-LRJWQT2R.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/unified-portable-profile/unified-profile/manifest.ts
var unifiedPortableProfileUnifiedProfileAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: UnifiedPortableProfileUnifiedProfileAriResourceOwner,
  resourceType: UnifiedPortableProfileUnifiedProfileAriResourceType,
  resourceIdSlug: "{unifiedProfileInternalId}",
  resourceIdSegmentFormats: {
    unifiedProfileInternalId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-portable-profile/unified-profile/index.ts
var UnifiedPortableProfileUnifiedProfileAri = class _UnifiedPortableProfileUnifiedProfileAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._unifiedProfileInternalId = opts.resourceIdSegmentValues.unifiedProfileInternalId;
  }
  get unifiedProfileInternalId() {
    return this._unifiedProfileInternalId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedPortableProfileUnifiedProfileAriStaticOpts.qualifier,
      platformQualifier: unifiedPortableProfileUnifiedProfileAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedPortableProfileUnifiedProfileAriStaticOpts.resourceOwner,
      resourceType: unifiedPortableProfileUnifiedProfileAriStaticOpts.resourceType,
      resourceId: `${opts.unifiedProfileInternalId}`,
      resourceIdSegmentValues: {
        unifiedProfileInternalId: opts.unifiedProfileInternalId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedPortableProfileUnifiedProfileAriStaticOpts);
    return new _UnifiedPortableProfileUnifiedProfileAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedPortableProfileUnifiedProfileAriStaticOpts);
    return new _UnifiedPortableProfileUnifiedProfileAri(opts);
  }
  getVariables() {
    return {
      unifiedProfileInternalId: this.unifiedProfileInternalId
    };
  }
};

export {
  UnifiedPortableProfileUnifiedProfileAri
};
