// @forge/util/packages/ari/chunk-5EAMMOK3.mjs
import {
  GoogleDocumentAriResourceOwner,
  GoogleDocumentAriResourceType
} from "./chunk-5VDZQSQU.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/google/document/manifest.ts
var googleDocumentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: GoogleDocumentAriResourceOwner,
  resourceType: GoogleDocumentAriResourceType,
  resourceIdSlug: "{docId}",
  resourceIdSegmentFormats: {
    docId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/google/document/index.ts
var GoogleDocumentAri = class _GoogleDocumentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._docId = opts.resourceIdSegmentValues.docId;
  }
  get docId() {
    return this._docId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleDocumentAriStaticOpts.qualifier,
      platformQualifier: googleDocumentAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleDocumentAriStaticOpts.resourceOwner,
      resourceType: googleDocumentAriStaticOpts.resourceType,
      resourceId: `${opts.docId}`,
      resourceIdSegmentValues: {
        docId: opts.docId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleDocumentAriStaticOpts);
    return new _GoogleDocumentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleDocumentAriStaticOpts);
    return new _GoogleDocumentAri(opts);
  }
  getVariables() {
    return {
      docId: this.docId
    };
  }
};

export {
  GoogleDocumentAri
};
