// @forge/util/packages/ari/chunk-UJDKKKW4.mjs
import {
  PostOfficeMessageInstanceAriResourceOwner,
  PostOfficeMessageInstanceAriResourceType
} from "./chunk-6N62CYFU.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/post-office/message-instance/manifest.ts
var postOfficeMessageInstanceAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: PostOfficeMessageInstanceAriResourceOwner,
  resourceType: PostOfficeMessageInstanceAriResourceType,
  resourceIdSlug: "{messageTemplateId}/{messageInstanceId}",
  resourceIdSegmentFormats: {
    messageTemplateId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    messageInstanceId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/post-office/message-instance/index.ts
var PostOfficeMessageInstanceAri = class _PostOfficeMessageInstanceAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._messageTemplateId = opts.resourceIdSegmentValues.messageTemplateId, this._messageInstanceId = opts.resourceIdSegmentValues.messageInstanceId;
  }
  get messageTemplateId() {
    return this._messageTemplateId;
  }
  get messageInstanceId() {
    return this._messageInstanceId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: postOfficeMessageInstanceAriStaticOpts.qualifier,
      platformQualifier: postOfficeMessageInstanceAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: postOfficeMessageInstanceAriStaticOpts.resourceOwner,
      resourceType: postOfficeMessageInstanceAriStaticOpts.resourceType,
      resourceId: `${opts.messageTemplateId}/${opts.messageInstanceId}`,
      resourceIdSegmentValues: {
        messageTemplateId: opts.messageTemplateId,
        messageInstanceId: opts.messageInstanceId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, postOfficeMessageInstanceAriStaticOpts);
    return new _PostOfficeMessageInstanceAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, postOfficeMessageInstanceAriStaticOpts);
    return new _PostOfficeMessageInstanceAri(opts);
  }
  getVariables() {
    return {
      messageTemplateId: this.messageTemplateId,
      messageInstanceId: this.messageInstanceId
    };
  }
};

export {
  PostOfficeMessageInstanceAri
};
