// @forge/util/packages/ari/chunk-FIF3NGCF.mjs
import {
  JiraCoreRoleAriResourceOwner,
  JiraCoreRoleAriResourceType
} from "./chunk-22MNSNE6.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/jira-core/role/manifest.ts
var jiraCoreRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: JiraCoreRoleAriResourceOwner,
  resourceType: JiraCoreRoleAriResourceType,
  resourceIdSlug: "{roleId}",
  resourceIdSegmentFormats: {
    roleId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira-core/role/index.ts
var JiraCoreRoleAri = class _JiraCoreRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleId = opts.resourceIdSegmentValues.roleId;
  }
  get roleId() {
    return this._roleId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraCoreRoleAriStaticOpts.qualifier,
      platformQualifier: jiraCoreRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: jiraCoreRoleAriStaticOpts.resourceOwner,
      resourceType: jiraCoreRoleAriStaticOpts.resourceType,
      resourceId: `${opts.roleId}`,
      resourceIdSegmentValues: {
        roleId: opts.roleId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraCoreRoleAriStaticOpts);
    return new _JiraCoreRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraCoreRoleAriStaticOpts);
    return new _JiraCoreRoleAri(opts);
  }
  getVariables() {
    return {
      roleId: this.roleId
    };
  }
};

export {
  JiraCoreRoleAri
};
