// @forge/util/packages/ari/chunk-YUATWIVK.mjs
import {
  MarketplaceTeamAriResourceOwner,
  MarketplaceTeamAriResourceType
} from "./chunk-IUUUDGF6.mjs";
import {
  RegisteredAri
} from "./chunk-CLQNXEWF.mjs";
import {
  AriParser
} from "./chunk-HNZGDVUS.mjs";

// src/marketplace/team/manifest.ts
var marketplaceTeamAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: MarketplaceTeamAriResourceOwner,
  resourceType: MarketplaceTeamAriResourceType,
  resourceIdSlug: "{teamId}",
  resourceIdSegmentFormats: {
    teamId: /[a-zA-Z0-9.\-_~]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/marketplace/team/index.ts
var MarketplaceTeamAri = class _MarketplaceTeamAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._teamId = opts.resourceIdSegmentValues.teamId;
  }
  get teamId() {
    return this._teamId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: marketplaceTeamAriStaticOpts.qualifier,
      platformQualifier: marketplaceTeamAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: marketplaceTeamAriStaticOpts.resourceOwner,
      resourceType: marketplaceTeamAriStaticOpts.resourceType,
      resourceId: `${opts.teamId}`,
      resourceIdSegmentValues: {
        teamId: opts.teamId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, marketplaceTeamAriStaticOpts);
    return new _MarketplaceTeamAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, marketplaceTeamAriStaticOpts);
    return new _MarketplaceTeamAri(opts);
  }
  getVariables() {
    return {
      teamId: this.teamId
    };
  }
};

export {
  MarketplaceTeamAri
};
