// @forge/util/packages/ari/workday/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);

// src/workday/index.ts
var workday_exports = {};
__export(workday_exports, {
  WorkdayCostCenterAri: () => WorkdayCostCenterAri,
  WorkdayJobProfileAri: () => WorkdayJobProfileAri,
  WorkdayPositionAri: () => WorkdayPositionAri,
  WorkdayStaffAri: () => WorkdayStaffAri,
  WorkdaySupervisoryOrgAri: () => WorkdaySupervisoryOrgAri,
  WorkdayWorkerAri: () => WorkdayWorkerAri
});
module.exports = __toCommonJS(workday_exports);

// src/errors.ts
var ValidationError = class extends Error {
};

// src/core/rules/cloud-id.ts
function validateCloudId(cloudId, format = new RegExp("^[a-zA-Z0-9\\-]*$")) {
  if (!cloudId.match(format))
    throw new ValidationError(`Invalid cloud ID, expected ID of format ${format}.`);
}

// src/core/rules/platform-qualifier.ts
function validatePlatformQualifier(qualifier) {
  if (qualifier !== "cloud" && qualifier !== "third-party")
    throw new ValidationError(`Identifier must have a qualifier of 'cloud' or 'third-party'. Received: ${qualifier}`);
}

// src/core/rules/qualifier.ts
function validateAtiQualifier(qualifier) {
  if (qualifier !== "ati")
    throw new ValidationError(`ATI must have a qualifier of 'ati'. Received: ${qualifier}`);
}
function validateAriQualifier(qualifier) {
  if (qualifier !== "ari")
    throw new ValidationError(`ARI must have a qualifier of 'ari'. Received: ${qualifier}`);
}

// src/core/rules/resource-id.ts
function validateResourceId(id, format = new RegExp("[!a-zA-Z0-9\\-_.~@:{}=]+(/[!a-zA-Z0-9\\-_.~@:{}=]+)*" /* ANY_RESOURCE_ID */), key = "resourceId") {
  let formatWithCarets = new RegExp(`^${format.source}$`);
  if (!id.match(formatWithCarets))
    throw new ValidationError(`Invalid ${key} - ${id}, expected ID of format ${formatWithCarets}.`);
}

// src/core/rules/resource-id-segments.ts
function validateResourceIdSegments(resourceIdSegmentValues, resourceIdSegmentFormats) {
  Object.entries(resourceIdSegmentValues).forEach(([resourceIdKey, resourceIdValue]) => {
    validateResourceId(resourceIdValue, resourceIdSegmentFormats == null ? void 0 : resourceIdSegmentFormats[resourceIdKey], resourceIdKey);
  });
}

// src/core/rules/resource-owner.ts
function validateResourceOwner(owner, expectedResourceOwner) {
  if (expectedResourceOwner && owner !== expectedResourceOwner)
    throw new ValidationError(`Invalid resource owner - ${owner}, expected ${expectedResourceOwner}.`);
}

// src/core/rules/resource-type.ts
function validateResourceType(type, expectedResourceType) {
  if (expectedResourceType && type !== expectedResourceType)
    throw new ValidationError(`Invalid resource type - ${type}, expected ${expectedResourceType}.`);
}

// src/core/parser/base.ts
var NUMBER_OF_BASE_SEGMENTS = 5, SEGMENT_SEPARATOR = ":", BaseParser = class {
  static getIdentifierSegments(inputStr, numberOfSegmentsRequired) {
    let allSegments = inputStr.split(SEGMENT_SEPARATOR);
    if (allSegments.length < numberOfSegmentsRequired)
      throw new ValidationError(`Input string must have ${numberOfSegmentsRequired} segments.`);
    if (allSegments.length > numberOfSegmentsRequired) {
      let segments = allSegments.slice(0, NUMBER_OF_BASE_SEGMENTS - 1), segmentsForResourceId = allSegments.slice(NUMBER_OF_BASE_SEGMENTS - 1, allSegments.length);
      return [...segments, segmentsForResourceId.join(SEGMENT_SEPARATOR)];
    }
    return allSegments;
  }
};

// src/core/parser/ari.ts
var NUMBER_OF_ARI_SEGMENTS = 5, AriParser = class extends BaseParser {
  static fromString(maybeAri, ariProperties) {
    let segments = this.getIdentifierSegments(maybeAri, NUMBER_OF_ARI_SEGMENTS), [qualifier, platformQualifier, resourceOwner, cloudId, resourceTypeAndId] = segments, [resourceType, ...resourceIdSegments] = resourceTypeAndId.split("/"), maybeAriDerivedProperties = {
      qualifier,
      platformQualifier,
      cloudId,
      resourceOwner,
      resourceType,
      resourceId: resourceIdSegments.join("/"),
      resourceIdSegmentValues: ariProperties != null && ariProperties.resourceIdSlug ? this.ariSegmentValuesFromSlug(
        ariProperties.resourceIdSlug,
        resourceIdSegments.join("/"),
        Object.keys(ariProperties.resourceIdSegmentFormats)
      ) : {}
    };
    return this.fromOpts(maybeAriDerivedProperties, ariProperties);
  }
  static fromOpts(maybeAriOpts, ariProperties) {
    let {
      qualifier = "ari",
      platformQualifier = "cloud",
      resourceOwner,
      cloudId,
      resourceId,
      resourceIdSegmentValues,
      resourceType
    } = maybeAriOpts;
    return validateAriQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateCloudId(maybeAriOpts.cloudId || "", ariProperties == null ? void 0 : ariProperties.cloudId), validateResourceOwner(resourceOwner, ariProperties == null ? void 0 : ariProperties.resourceOwner), validateResourceType(resourceType, ariProperties == null ? void 0 : ariProperties.resourceType), validateResourceIdSegments(resourceIdSegmentValues, ariProperties == null ? void 0 : ariProperties.resourceIdSegmentFormats), {
      qualifier,
      resourceOwner,
      resourceType,
      platformQualifier,
      cloudId,
      resourceId,
      resourceIdSegmentValues
    };
  }
  static ariSegmentValuesFromSlug(slug, incomingSegments, expectedKeys) {
    let regexpResult = new RegExp("^" + slug.replace(/\{(.*?)\}/g, "(?<$1>.*?)") + "$").exec(incomingSegments);
    if (!regexpResult)
      throw new ValidationError(`Segment '${incomingSegments}' don't match expected slug: ${slug}`);
    return expectedKeys.forEach((expectedKey) => {
      var _a;
      if (!((_a = regexpResult.groups) != null && _a[expectedKey]))
        throw new ValidationError(`No value supplied for '${expectedKey}' based on slug ${slug}`);
    }), regexpResult.groups || {};
  }
};

// src/core/parser/ati.ts
var NUMBER_OF_ATI_SEGMENTS = 4, AtiParser = class extends BaseParser {
  static fromString(inputStr, atiOpts) {
    let segments = this.getIdentifierSegments(inputStr, NUMBER_OF_ATI_SEGMENTS), [qualifier = "ati", platformQualifier = "cloud", resourceOwner, resourceType] = segments;
    return validateAtiQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateResourceOwner(resourceOwner, atiOpts == null ? void 0 : atiOpts.resourceOwner), validateResourceType(resourceType, atiOpts == null ? void 0 : atiOpts.resourceType), { platformQualifier, resourceOwner, resourceType };
  }
  static fromOpts(maybeAtiOpts, atiOpts) {
    let { qualifier, platformQualifier, resourceOwner, resourceType } = maybeAtiOpts;
    return validateAtiQualifier(qualifier), validatePlatformQualifier(platformQualifier), validateResourceOwner(resourceOwner, atiOpts == null ? void 0 : atiOpts.resourceOwner), validateResourceType(resourceType, atiOpts == null ? void 0 : atiOpts.resourceType), {
      platformQualifier,
      resourceOwner,
      resourceType
    };
  }
};

// src/ati.ts
var Ati = class _Ati {
  constructor(platformQualifier, resourceOwner, resourceType) {
    this._platformQualifier = platformQualifier, this._resourceOwner = resourceOwner, this._resourceType = resourceType;
  }
  get platformQualifier() {
    return this._platformQualifier;
  }
  get resourceOwner() {
    return this._resourceOwner;
  }
  get resourceType() {
    return this._resourceType;
  }
  static create(opts) {
    return new _Ati(opts.platformQualifier || "cloud", opts.resourceOwner, opts.resourceType);
  }
  static parse(maybeAti, atiOpts) {
    let opts = AtiParser.fromString(maybeAti, atiOpts);
    return new _Ati(opts.platformQualifier, opts.resourceOwner, opts.resourceType);
  }
  toString() {
    return `ati:${this.platformQualifier}:${this.resourceOwner}:${this.resourceType}`;
  }
  toJSON() {
    return this.toString();
  }
  toOpts() {
    return {
      platformQualifier: this._platformQualifier,
      resourceOwner: this._resourceOwner,
      resourceType: this._resourceType
    };
  }
};

// src/core/ari.ts
var Ari = class {
  constructor(opts) {
    this._ati = Ati.create({
      platformQualifier: opts.platformQualifier || "cloud",
      resourceOwner: opts.resourceOwner,
      resourceType: opts.resourceType
    }), this._cloudId = opts.cloudId !== "" ? opts.cloudId : void 0, this._resourceId = opts.resourceId;
  }
  get platformQualifier() {
    return this._ati.platformQualifier;
  }
  get cloudId() {
    return this._cloudId;
  }
  get resourceOwner() {
    return this._ati.resourceOwner;
  }
  get resourceType() {
    return this._ati.resourceType;
  }
  get resourceId() {
    return this._resourceId;
  }
  get ati() {
    return this._ati;
  }
  equals(other) {
    return this.toString() === other.toString();
  }
  toString() {
    return `ari:${this.platformQualifier}:${this.resourceOwner}:${this.cloudId || ""}:${this.resourceType}/${this.resourceId}`;
  }
  toJSON() {
    return this.toString();
  }
  toOpts() {
    return {
      platformQualifier: this.platformQualifier,
      resourceOwner: this.resourceOwner,
      cloudId: this.cloudId,
      resourceType: this.resourceType,
      resourceId: this.resourceId
    };
  }
};

// src/any-ari.ts
var AnyAri = class _AnyAri extends Ari {
  constructor(opts) {
    super(opts);
  }
  static create(ariOpts, ariStaticOpts) {
    let ariOptsWithDefaults = { ...ariOpts, resourceIdSegmentValues: {} }, validatedOpts = AriParser.fromOpts(ariOptsWithDefaults, ariStaticOpts);
    return new _AnyAri(validatedOpts);
  }
  static parse(maybeAri, ariStaticOpts) {
    let validatedOpts = AriParser.fromString(maybeAri, ariStaticOpts);
    return new _AnyAri(validatedOpts);
  }
  static check(maybeAri) {
    try {
      return _AnyAri.parse(maybeAri.toString()), !0;
    } catch (err) {
      return !1;
    }
  }
  asAnyAri() {
    return this;
  }
};

// src/core/registered-ari.ts
var RegisteredAri = class extends Ari {
  static check(maybeAri) {
    try {
      return this.parse(maybeAri.toString()), !0;
    } catch (err) {
      return !1;
    }
  }
  asAnyAri() {
    return AnyAri.parse(this.toString());
  }
};

// src/workday/cost-center/types.ts
var WorkdayCostCenterAriResourceOwner = "workday", WorkdayCostCenterAriResourceType = "cost-center";

// src/workday/cost-center/manifest.ts
var workdayCostCenterAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayCostCenterAriResourceOwner,
  resourceType: WorkdayCostCenterAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{costCenterWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    costCenterWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/cost-center/index.ts
var WorkdayCostCenterAri = class _WorkdayCostCenterAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._costCenterWorkdayId = opts.resourceIdSegmentValues.costCenterWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get costCenterWorkdayId() {
    return this._costCenterWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayCostCenterAriStaticOpts.qualifier,
      platformQualifier: workdayCostCenterAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayCostCenterAriStaticOpts.resourceOwner,
      resourceType: workdayCostCenterAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.costCenterWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        costCenterWorkdayId: opts.costCenterWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayCostCenterAriStaticOpts);
    return new _WorkdayCostCenterAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayCostCenterAriStaticOpts);
    return new _WorkdayCostCenterAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      costCenterWorkdayId: this.costCenterWorkdayId
    };
  }
};

// src/workday/job-profile/types.ts
var WorkdayJobProfileAriResourceOwner = "workday", WorkdayJobProfileAriResourceType = "job-profile";

// src/workday/job-profile/manifest.ts
var workdayJobProfileAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayJobProfileAriResourceOwner,
  resourceType: WorkdayJobProfileAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{jobProfileWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    jobProfileWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/job-profile/index.ts
var WorkdayJobProfileAri = class _WorkdayJobProfileAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._jobProfileWorkdayId = opts.resourceIdSegmentValues.jobProfileWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get jobProfileWorkdayId() {
    return this._jobProfileWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayJobProfileAriStaticOpts.qualifier,
      platformQualifier: workdayJobProfileAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayJobProfileAriStaticOpts.resourceOwner,
      resourceType: workdayJobProfileAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.jobProfileWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        jobProfileWorkdayId: opts.jobProfileWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayJobProfileAriStaticOpts);
    return new _WorkdayJobProfileAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayJobProfileAriStaticOpts);
    return new _WorkdayJobProfileAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      jobProfileWorkdayId: this.jobProfileWorkdayId
    };
  }
};

// src/workday/position/types.ts
var WorkdayPositionAriResourceOwner = "workday", WorkdayPositionAriResourceType = "position";

// src/workday/position/manifest.ts
var workdayPositionAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayPositionAriResourceOwner,
  resourceType: WorkdayPositionAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{positionWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    positionWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/position/index.ts
var WorkdayPositionAri = class _WorkdayPositionAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._positionWorkdayId = opts.resourceIdSegmentValues.positionWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get positionWorkdayId() {
    return this._positionWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayPositionAriStaticOpts.qualifier,
      platformQualifier: workdayPositionAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayPositionAriStaticOpts.resourceOwner,
      resourceType: workdayPositionAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.positionWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        positionWorkdayId: opts.positionWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayPositionAriStaticOpts);
    return new _WorkdayPositionAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayPositionAriStaticOpts);
    return new _WorkdayPositionAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      positionWorkdayId: this.positionWorkdayId
    };
  }
};

// src/workday/staff/types.ts
var WorkdayStaffAriResourceOwner = "workday", WorkdayStaffAriResourceType = "staff";

// src/workday/staff/manifest.ts
var workdayStaffAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayStaffAriResourceOwner,
  resourceType: WorkdayStaffAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{staffId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    staffId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/staff/index.ts
var WorkdayStaffAri = class _WorkdayStaffAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._staffId = opts.resourceIdSegmentValues.staffId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get staffId() {
    return this._staffId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayStaffAriStaticOpts.qualifier,
      platformQualifier: workdayStaffAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayStaffAriStaticOpts.resourceOwner,
      resourceType: workdayStaffAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.staffId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        staffId: opts.staffId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayStaffAriStaticOpts);
    return new _WorkdayStaffAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayStaffAriStaticOpts);
    return new _WorkdayStaffAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      staffId: this.staffId
    };
  }
};

// src/workday/supervisory-org/types.ts
var WorkdaySupervisoryOrgAriResourceOwner = "workday", WorkdaySupervisoryOrgAriResourceType = "supervisory-org";

// src/workday/supervisory-org/manifest.ts
var workdaySupervisoryOrgAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdaySupervisoryOrgAriResourceOwner,
  resourceType: WorkdaySupervisoryOrgAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{supervisoryOrgWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    supervisoryOrgWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/supervisory-org/index.ts
var WorkdaySupervisoryOrgAri = class _WorkdaySupervisoryOrgAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._supervisoryOrgWorkdayId = opts.resourceIdSegmentValues.supervisoryOrgWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get supervisoryOrgWorkdayId() {
    return this._supervisoryOrgWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdaySupervisoryOrgAriStaticOpts.qualifier,
      platformQualifier: workdaySupervisoryOrgAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdaySupervisoryOrgAriStaticOpts.resourceOwner,
      resourceType: workdaySupervisoryOrgAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.supervisoryOrgWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        supervisoryOrgWorkdayId: opts.supervisoryOrgWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdaySupervisoryOrgAriStaticOpts);
    return new _WorkdaySupervisoryOrgAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdaySupervisoryOrgAriStaticOpts);
    return new _WorkdaySupervisoryOrgAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      supervisoryOrgWorkdayId: this.supervisoryOrgWorkdayId
    };
  }
};

// src/workday/worker/types.ts
var WorkdayWorkerAriResourceOwner = "workday", WorkdayWorkerAriResourceType = "worker";

// src/workday/worker/manifest.ts
var workdayWorkerAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "third-party",
  cloudId: new RegExp("^$"),
  resourceOwner: WorkdayWorkerAriResourceOwner,
  resourceType: WorkdayWorkerAriResourceType,
  resourceIdSlug: "{workdayTenantKey}/{workerWorkdayId}",
  resourceIdSegmentFormats: {
    workdayTenantKey: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/,
    // eslint-disable-line no-useless-escape
    workerWorkdayId: /[!a-zA-Z0-9\-_.~@:{}=]+(\/[!a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/workday/worker/index.ts
var WorkdayWorkerAri = class _WorkdayWorkerAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._workdayTenantKey = opts.resourceIdSegmentValues.workdayTenantKey, this._workerWorkdayId = opts.resourceIdSegmentValues.workerWorkdayId;
  }
  get workdayTenantKey() {
    return this._workdayTenantKey;
  }
  get workerWorkdayId() {
    return this._workerWorkdayId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: workdayWorkerAriStaticOpts.qualifier,
      platformQualifier: workdayWorkerAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: workdayWorkerAriStaticOpts.resourceOwner,
      resourceType: workdayWorkerAriStaticOpts.resourceType,
      resourceId: `${opts.workdayTenantKey}/${opts.workerWorkdayId}`,
      resourceIdSegmentValues: {
        workdayTenantKey: opts.workdayTenantKey,
        workerWorkdayId: opts.workerWorkdayId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, workdayWorkerAriStaticOpts);
    return new _WorkdayWorkerAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, workdayWorkerAriStaticOpts);
    return new _WorkdayWorkerAri(opts);
  }
  getVariables() {
    return {
      workdayTenantKey: this.workdayTenantKey,
      workerWorkdayId: this.workerWorkdayId
    };
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WorkdayCostCenterAri,
  WorkdayJobProfileAri,
  WorkdayPositionAri,
  WorkdayStaffAri,
  WorkdaySupervisoryOrgAri,
  WorkdayWorkerAri
});
